---
title: "Analysis of the gut microbiome during pregnancy in T1D 16S rRNA (MiSeq)"
author: "Alexandra Roth Schulze"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    number_sections: yes
    toc_float: yes
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
    code_folding: "hide"
urlcolor: blue
---

```{r echo=FALSE}
options(width = 90)
```

```{r ,include=FALSE}
library(knitr)
opts_chunk$set(comment = NA, warning=FALSE, message=FALSE, echo=TRUE, tidy=TRUE, tidy.opts=list(width.cutoff=60))
library("ggplot2")
theme_set(theme_bw())
pal = "Set1"
scale_colour_discrete <- function(palname = pal, ...) {
    scale_colour_brewer(palette = palname, ...)
}
scale_fill_discrete <- function(palname = pal, ...) {
    scale_fill_brewer(palette = palname, ...)
}
```


```{r PackageLoad, tidy=TRUE, message=FALSE, warning=FALSE, echo=FALSE}
library("ggplot2", lib.loc="~/Library/R/3.4/library")
library("ade4", lib.loc="~/Library/R/3.5/library")
library("plotly")
library("RColorBrewer")
library("matrixStats")
library("knitr")
library("RColorBrewer")
library("limma")
library("edgeR")
library("matrixStats")
library("phyloseq")
library("vegan")
library("reshape2")
library("scales")
library("lme4")
library("lmerTest")
library("geepack")
library("dplyr")
library("Hmisc")
library("emmeans")
library("doBy")
library("data.table")
library("microbiomeSeq")
library("devtools")
library("adespatial")
library("genefilter")
library("metagMisc")
library("easyGgplot2")
library("stringr")
```

```{r RMAPfunction, echo=FALSE}
## Function to perform PERMANOVA analysis with repeat measure-aware permutations 
PERMANOVA_repeat_measures <- function(
  D, permute_within, blocks = NULL, block_data, permutations=999,
  metadata_order = c(names(permute_within), names(block_data)),
  na.rm=F) {
  
  # Make sure D is a dist object
  if (class(D) != "dist") {
    stop("D must be a dist object")
  }
  
  # Default to free permutations if blocks is not given
  if (!missing(block_data) && is.null(blocks)) {
    stop("blocks must be given if block_data is present")
  } else if (is.null(blocks)) {
    blocks <- rep(1, nrow(permute_within))
    block_data <- as.data.frame(matrix(0, nrow=1, ncol=0))
  } else if (length(unique(blocks)) == 1) {
    warning("blocks only contains one unique value")
  }
  
  # Ensure no metadata overlap between permute_within and block_data
  if (length(intersect(names(permute_within), names(block_data))) > 0) {
    stop("metadata is repeated across permute_within and block_data")
  }
  
  # Ensure that metadata_order only contains stuff in permute_within and block_data
  if(length(setdiff(metadata_order, union(names(permute_within), names(block_data)))) > 0) {
    stop("metadata_order contains metadata not in permute_within and block_data")
  }
  
  # Ensure that the data in permute_within matches that in dist
  ord <- rownames(as.matrix(D))
  if (length(ord) != nrow(permute_within) || length(blocks) != length(ord)) {
    stop("blocks, permute_within, and D are not the same size")
  }
  if (is.null(rownames(permute_within))) {
    warning("permute_within has no rownames - can't verify sample orders")
  } else if (!all(ord == rownames(permute_within))) {
    stop("rownames do not match between permute_within and D")
  }
  
  # Ensure matching between blocks and block_data
  if (any(is.na(blocks))) {
    stop("NAs are not allowed in blocks")
  }
  if (is.factor(blocks)) {
    if (length(levels(blocks)) != nrow(block_data)) {
      stop("block_data does not have as many rows as blocks has levels")
    }
    if (!is.null(rownames(block_data)) && any(rownames(block_data) != levels(blocks))) {
      stop("block_data rownames does not match the levels of blocks")
    }
    # Discard level information
    blocks <- as.numeric(blocks)
  } else if (is.numeric(blocks)) {
    if (blocks < 1 || max(blocks) > nrow(block_data)) {
      stop("Numeric blocks has indices out of range")
    }
  } else if (is.character(blocks)) {
    if (is.null(rownames(block_data)) || !all(blocks %in% rownames(block_data))) {
      stop("blocks does not match the rownames of block_data")
    }
    # Transform to numeric
    blocks <- match(blocks, rownames(block_data))
  } else {
    stop("blocks must be a numeric, factor, or character vector")
  }
  
  # Error out on NA metadata rather than allowing adonis to error out with
  # a totally nonsensical error message
  if (any(is.na(permute_within)) || any(is.na(block_data))) {
    if (na.rm) {
      n_prerm <- length(blocks)
      
      # Remove NAs in block_data
      hasna <- (rowSums(is.na(block_data)) > 0) | (sapply(split(rowSums(is.na(permute_within)) > 0, blocks), mean) == 1)
      block_data <- block_data[!hasna,, drop=F]
      keep <- !hasna[blocks]
      blocks <- cumsum(!hasna)[blocks]
      
      blocks <- blocks[keep]
      permute_within <- permute_within[keep,, drop=F]
      D <- as.matrix(D)[keep, keep]
      # block_data is not subset, as the rows with NAs are no longer referenced in blocks
      
      # Remove NAs in permute_within
      keep <- rowSums(is.na(permute_within)) == 0
      blocks <- blocks[keep]
      permute_within <- permute_within[keep,, drop=F]
      D <- as.dist(D[keep, keep])
      
      if (length(blocks) < ncol(permute_within) + ncol(block_data)) {
        stop(sprintf("After omitting samples with NAs, the number of samples (%d) is less than the number of metadata (%d)",
                     length(blocks), ncol(permute_within) + ncol(block_data)))
      } else if (length(blocks) < n_prerm * 0.5) {
        warning(sprintf("Removed %d samples with NA metadata", n_prerm - length(blocks)))
      }
    } else {
      stop("Some metadata is NA! adonis does not support any NA in the metadata")
    }
  }
  
  # Warn on some suspicious input
  persample <- apply(permute_within, 1, function(x)is.factor(x) && !any(duplicated(x)))
  if (any(persample)) {
    warning(sprintf("%s in permute_within has one DOF per sample.", colnames(permute_within)[which(persample)[1]]))
  }
  if (length(unique(blocks)) < nrow(block_data)) {
    warning("Not all blocks have a sample associated with them. Block permutations will still be performed over the full set of blocks - if this is not desired, subset block_data to only the blocks which appear in the data.")
  }
  if (!any(duplicated(blocks))) {
    warning("blocks contains no duplicated elements")
  }
  
  library(vegan)
  library(permute)
  
  # Test statistic from non-permuted data
  mtdat <- cbind(permute_within, block_data[blocks,,drop=F])
  ad <- adonis(D ~ ., permutations=0, data=mtdat[, metadata_order, drop=F])
  R2 <- ad$aov.tab$R2
  names(R2) <- rownames(ad$aov.tab)
  
  # Permutations
  nullsamples <- matrix(NA, nrow=length(R2), ncol=permutations)
  for (i in seq_len(permutations)) {
    within.i <- shuffle(nrow(permute_within), control=how(blocks=blocks))
    block.i <- sample(seq_len(nrow(block_data)))
    mtdat <- cbind(
      permute_within[within.i,,drop=F],
      block_data[block.i,,drop=F][blocks,,drop=F])
    perm.ad <- adonis(D ~ ., permutations=0, data=mtdat[, metadata_order, drop=F])
    
    nullsamples[,i] <- perm.ad$aov.tab$R2
  }
  
  # For residuals, test the other direction (i.e. p-value of all covariates)
  n <- length(R2)
  R2[n-1] <- 1 - R2[n-1]
  nullsamples[n-1,] <- 1 - nullsamples[n-1,]
  
  # P value calculation similar to adonis's
  exceedances <- rowSums(nullsamples > R2)
  P <- (exceedances + 1) / (permutations + 1)
  
  P[n] <- NA    # No p-values for "Total"
  ad$aov.tab$`Pr(>F)` <- P
  
  return (ad)
}
```

```{r StdErrorPlotfunction, echo=FALSE}
#For plotting abundances of Taxa across time for T1D and nonT1D

## Gives count, mean, standard deviation, standard error of the mean, and confidence interval (default 95%).
##   data: a data frame.
##   measurevar: the name of a column that contains the variable to be summariezed
##   groupvars: a vector containing names of columns that contain grouping variables
##   na.rm: a boolean that indicates whether to ignore NA's
##   conf.interval: the percent range of the confidence interval (default is 95%)
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r TaxaFun, echo=FALSE}
##Function for making nice names that contain the best classified level for each taxa in column 8
makeTaxLabel <- function(OTU, mydata){
    # Makes a string label using the lowest informative tax level
    #
    # Args:
    #   OTU: OTU number
    #   mydata: the phyloseq object with the tax table
    #
    # Returns:
    #   a tax name
    OTU <- as.character(OTU)  # the OTU numbers are stored as character not integer!
    taxstrings <- as.character(taxTab(mydata)[OTU])
    empty_strings <- c("k__", "p__", "c__", "o__", "f__", "g__", "s__")
    tax_name <- NA
    tax_level <- length(taxstrings)  # start at lowest tax level

    while(is.na(tax_name) | 
              (tax_name %in% empty_strings)){
        tax_name  <- taxstrings[tax_level]
        tax_level <- tax_level -1
    }
    tax_name
}
```


#Background

This report shows the analysis performed for 16S rRNA gene amplicon sequencing from the gut microbiome of 148 pregnant women. For each pregnant woman, samples were taken at different time points (i.e. from 1 to 3 different time points per woman) during the pregnancy (i.e. up until day ~280 when the child is born) to give a total of 377 samples. From the 148 pregnant women, 92 are diabetic (T1D) and 56 are healthy (nonT1D). Samples from the different women were NOT taken in the exact same Day in pregnancy, thus Day should be treated as a continuous variable rather than a categorical. However, time points could be categorised into trimesters if necessary, correspondence being:

Trimester 1 = 0-99 days

Trimester 2 = 100-196 days

Trimester 3 = 196-280 days

**Summary:**

145 pregnancies: 94 in T1D women and nonT1D=56 in wome without T1D (non-T1D).

A total of 139 women donated samples, 6 women donated samples from 2 different pregnancies giving a total a 145 pregnancies.

1-3 time points per woman per pregnancy (from day 37 to day 274 of the pregnancy).

354 samples in total: T1D=235 and nonT1D=119.

For the analysis we would explored if there is a difference in Alpha and Beta diversity between T1D and nonT1D. Furthermore, we would also identified differentially abundant operational taxonomic units (i.e. OTUs or sequence variants of the 16S rRNA that are roughly equivalent to a bacterial species) between T1D and nonT1D pregnant women.

**Sample details information:**

Samples were generated by PCR amplification from stool samples targeting the V4 region of the 16S rRNA gene. Each sample was sequenced twice on different sequence wells (i.e. technical sequencing replicates) from the Illumina MiSeq instrument with the paired-end 600-cycle (2 × 300) kit. Samples were sequenced in 6 different sequencing runs. Samples from the same pregnant woman at different time points were sequenced in the same run.

#Data generation

##Processing the sequences: from sequence data to feature table

The data was processed through Qiime2 (https://qiime2.org/) which is a next-generation microbiome bioinformatics platform written in python. The analysis starts with raw DNA sequencing data with the following steps and python commands that were performed for each sequencing runs separately:

1) Import paired-end sequencing data into qiime2.
  * command: qiime tools import --type EMPPairedEndSequences --input-path ./1-Run/ --output-path ./1-Run/1-Run_MISEQXXXX.qza
  
2) Demultiplex data to know which sample each paired-end reads belong to.
  * command: qiime demux emp-paired --m-barcodes-file metadata.tsv --m-barcodes-category BarcodeSequence --i-seqs 1-Run_MISEQXXX.qza --o-per-sample-sequences demux --p-rev-comp-mapping-barcodes
  
3) Trimming PCR primers: Search demultiplexed paired-end sequences for adapters and remove them. In this step everything from the amplification primers on is removed.
  * command: qiime cutadapt trim-paired --i-demultiplexed-sequences ./demux.qza --p-front-f GGACTACNVGGGTWTCTAAT --p-front-r GTGYCAGCMGCCGCGGTAA --o-trimmed-sequences trimmed_demux.qza --verbose
  
  The demultiplexed data resulting from this step was uploaded to the short read archive (SRA) database (https://www.ncbi.nlm.nih.gov/sra/) with accession number **PRJNA604850**.
  
4) Sequence quality control and feature table construction. In this step qiime2 uses the software Divisive Amplicon Denoising Algorithm 2 (DADA2; https://www.nature.com/articles/nmeth.3869), a quality-aware model-based approach for correcting Illumina amplicon errors. Sample composition is inferred by dividing amplicon reads into partitions consistent with the error model. DADA2 does not create OTUs (i.e. cluster sequences to a given % identity threshold) and does differentiate closely related ribosomal sequence variants (i.e. RSV or features). The complete DADA2 pipeline integrated in qiime2 implements sample inference, chimera identification, merging of paired-end reads and the creation of a "feture-per-sample counts table". 

  * command: qiime dada2 denoise-paired --i-demultiplexed-seqs trimmed_demux.qza --o-table table --o-representative-sequences rep-seqs --p-trunc-len-f 230 --p-trunc-len-r 160 --p-max-ee 3 --p-n-threads 8
  
The 6 runs where processed with the same parameters, including trimming of the forward reads to 230 basepairs and the reverse reads to 160 bp (i.e. parameters --p-trunc-len-f 230 and --p-trunc-len-r 160) and number of expected errors higher than 3 (--p-max-ee 3). Although these parameters should be different for different sequencing runs (depending on their quality), we used the same parameters for the 6 runs (i.e. parameters from the poorest quality run were used) in order to be able to merge the resulting 6 feature tables into 1. The feature ids in the resulting tables are presented as hashes of the sequences defining each feature. The hash will always be the same for the same sequence so this allows feature tables to be merged across runs of this method. You should only merge tables if the exact same parameters are used for each run. 

5) The 6 tables were merged:

  * Command to merge feature tables: qiime feature-table merge --i-tables table_Mother_1-Run.qza --i-tables table_Mother_2-Run.qza --i-tables table_Mother_3-Run.qza --i-tables table_Mother_4-Run.qza --i-tables table_Mother_6-Run.qza --i-tables table_Mother_7-Run.qza --o-merged-table table_Mother_All-Runs.qza
  
  * Command to merge representative sequences: qiime feature-table merge-seqs --i-data rep-seq1.qza --i-data rep-seq2.qza --i-data rep-seq3.qza --i-data rep-seq4.qza --i-data rep-seq6.qza --i-data rep-seq7.qza  --o-merged-data rep-seq_Allmerged.qza
  
  Information on which samples were sequenced in which run can be found in supplementary excel file E0. If the processing using DADA2 (within Qiime2) needs to be repeated, sequences from the different sequencing runs must be processed separatelly and then the resulting tables and representative sequences must be merged previous to analysis with phyloseq. 
  
6) The taxonomic classification was done using the resulting representative sequences with the following Qiime2 command:

  * qiime feature-classifier classify-sklearn --i-classifier /home/$YourPath/Naive_Bayes_classifiers/silva-119-99-515-806-nb-classifier.qza --i-reads rep-seq_Allmerged.qza --o-classification taxonomy_AllMerged.qza
  
7) A phylogenetic tree was build within Qiime2 as follows:

  * qiime alignment mafft --i-sequences rep-seq_Allmerged.qza --o-alignment aligned-rep-seq_Allmerged.qza
  * qiime alignment mask --i-alignment aligned-rep-seq_Allmerged.qza --o-masked-alignment masked-aligned-rep-seq_Allmerged.qza
  * qiime phylogeny fasttree --i-alignment masked-aligned-rep-seq_Allmerged.qza --o-tree unrooted-tree-rep-seq_Allmerged.qza
  * qiime phylogeny midpoint-root --i-tree unrooted-tree-rep-seq_Allmerged.qza --o-rooted-tree rooted-tree-rep-seq_Allmerged.qza
  
Note: The phylogenetic tree was unzipped to integrated it into the phyloseq object in R. File name: Mother_All-Runs_Merged_tree.nwk
  
8) A biom file containing the feature table, taxonomy and metadata was obtained from qiime2 unzipping the generated feature table and taxonomy:

  a) After unzipping table_Mother_All-Runs.qza we obtained table_Mother_All-Runs.biom, this biom file was transformed usig the software biom into a text file and then again into the correct biom format (JSON):
    * biom convert -i table_Mother_All-Runs.biom -o table_Mother_All-Runs.txt --to-tsv
    * biom convert -i table_Mother_All-Runs.txt -o table_Mother_All-Runs.biom --table-type="OTU table" --to-json
    
  b) The file taxonomy_AllMerged.qza containing the taxonomy was unzipped (obtaining taxonomy_AllMerged.txt) and the header "#OTUID taxonomy confidence" was added. The format of the feature classification was also changed to obtain the greengenes format (e.g.           k__Bacteria; p__XXX; c__XX, etc). After this, the biom sortware was used to add the taxonomi to the biom file:
    *biom add-metadata --sc-separated taxonomy --observation-header OTUID,taxonomy --observation-metadata-fp taxonomy_AllMerged.txt -i table_Mother_All-Runs.biom -o table_Mother_All-Runs_wTax.biom
    
  c) The metadata (i.e. per sample information for each of the variables of factors of interest) was added:
    * biom add-metadata -i table_Mother_All-Runs_wTax.biom -o table_Mother_All-Runs_wTaxMet.biom --sample-metadata-fp ./metadata_All-RunsMerged.csv

  d) The resulting file in c) was used as input into phyloseq in order to de-identify (i.e. leave no information that could associate the samples to a person in the study) the samples prior to analysis, by replacing the metadata and saving into an R object "Mothers_OTU_AFilt1_Preg.RData".
  
##Processing, filtering and count normalization of feature table

The resulting R object containing the feature table, sample data and taxonomy (Mothers_OTU_AFilt1_Preg.RData), the and phylogenetic tree were imported into Phyloseq (https://github.com/joey711/phyloseq), an R package used for the preprocessing, analysis and graphical display of complex 16S rRNA amplicon data that has been already processed into a feature table. In this analysis phyloseq was used for agglomerate features into OTUs. This step was performed in a separate script: Phyloseq_Mother_data_formating_w7.R. Features that were less than 0.03% different at the sequence level were agglomerated into the same OTU based on the phylogenetic tree. After this, OTUs that had a relative abundance across all samples equal or less than 0.01% were removed. This left a total of 349 OTUs across all samples. This object was saved to Mothers_OTU_Phyloseq_Obj_Filter1_w7G.RData which was used for the analysis performed in this document.


```{r OTUs, echo=FALSE}
##IMPORTING DATA GENERATED in Phyloseq_Mother_data_formating_w7.R. 
Mothers_UnN_OTU <- load(file="~/$YOUR_PATH/Mothers_OTU_Phyloseq_Obj_Filter1_16S.RData")
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1

##Obtaining the best possible name to the lowest classification possible.
# Convert names that are not useful to NA
tax_table(Mothers_OTU_AFilt1_Preg) =gsub("uncultured organism", as.character(NA), tax_table(Mothers_OTU_AFilt1_Preg))
tax_table(Mothers_OTU_AFilt1_Preg) =gsub("uncultured bacterium", as.character(NA), tax_table(Mothers_OTU_AFilt1_Preg))
#Using function
mynames = NULL
for (i in 1:length(taxa_names(Mothers_OTU_AFilt1_Preg))){
mynames <- rbind(mynames, c(makeTaxLabel(taxa_names(Mothers_OTU_AFilt1_Preg)[i],Mothers_OTU_AFilt1_Preg)))
}
#mynames = str_c(mynames[,1],"_",taxa_names(Mothers_OTU_AFilt1_Preg))
tax_table(Mothers_OTU_AFilt1_Preg) <- cbind(tax_table(Mothers_OTU_AFilt1_Preg), OTUID=mynames)
colnames(tax_table(Mothers_OTU_AFilt1_Preg))[8] <- "Classification"

#Making sure each factor has the correct class

#This factor corresponds to extraction batch
sample_data(Mothers_OTU_AFilt1_Preg)$SeqRun <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$SeqRun)
#This factor corresponds to gestational age (day into pregnancy in which the sample was taken)
sample_data(Mothers_OTU_AFilt1_Preg)$Days <- as.integer(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$DaysG))
#This factor corresponds to conception age
sample_data(Mothers_OTU_AFilt1_Preg)$Age_LMP <- as.numeric(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$Age_LMP))
#This factor corresponds to Body Mass Index at conception time
sample_data(Mothers_OTU_AFilt1_Preg)$BMI_conception <- as.numeric(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$BMI_conception))
#This factor corresponds to Human leukocyte antigen (HLA) type of the women
sample_data(Mothers_OTU_AFilt1_Preg)$HLA.6DRML <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$HLA.6DRML)
#This factor corresponds to parity (which refers to if the women already had children before the current pregnancy)
sample_data(Mothers_OTU_AFilt1_Preg)$Nulliparous <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$Nulliparous)
#This factor corresponds to a unique identifier per person (i.e. if the woman contributed samples from different pregnancies, those samples belong to the same motherid)
sample_data(Mothers_OTU_AFilt1_Preg)$motherid <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$motherid)
#This factor corresponds to T1D status
sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus)
#This factor corresponds to trimester
sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)
#This factor corresponds to a composed factor of trimester and T1D status, which is used in the differential abundance analysis.
sample_data(Mothers_OTU_AFilt1_Preg)$TriwT1D <- factor(paste(sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr,sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus,sep="."))

#Original table saved in another variable to reset other steps
Original_OTU <- Mothers_OTU_AFilt1_Preg

# Subset to have only mothers trimester 1 
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T1"))
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg_T1, TakeOne%in%c("Y"))
SSumT1 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T1))
Mothers_OTU_AFilt1_Preg_T1_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T1, sample.size = SSumT1, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T1 <- Mothers_OTU_AFilt1_Preg_T1_RA

# Subset to have only mothers trimester 2 
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg_T2, TakeOne%in%c("Y"))
SSumT2 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T2))
Mothers_OTU_AFilt1_Preg_T2_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T2, sample.size = SSumT2, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T2 <- Mothers_OTU_AFilt1_Preg_T2_RA

# Subset to have only mothers trimester 3 
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg_T3, TakeOne%in%c("Y"))
SSumT3 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T3))
Mothers_OTU_AFilt1_Preg_T3_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T3, sample.size = SSumT3, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T3 <- Mothers_OTU_AFilt1_Preg_T3_RA

#Rarefy main dataset
SSum <- min(sample_sums(Mothers_OTU_AFilt1_Preg))
Mothers_OTU_AFilt1_Preg_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg, sample.size = SSum, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_RA

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))

#Subset to have only mothers without T1D
Mothers_OTU_AFilt1_PregwNT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("no"))

Mothers_OTU_AFilt1_Preg_T1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T2.yes"))
Mothers_OTU_AFilt1_Preg_T1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_T1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T2.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.no", "T3.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T3.no"))

TaxLv<- as.integer("8")
```

#Bar plots taxonomy

```{r BarPlot, echo=FALSE, fig.height=7, fig.width=14}
Personal_colors <- c("#FF7200", "#FF0000", "#FFAA00", "#FFDD00", "#72d813", "#154f0d", "#06993E", "#06D8C3", "#06B2D8", "#004ECC", "#0300cc", "#6200CC", "#8E00CC", "#C500CC", "#CC0073", "#CC002C", "#BA8857", "#A04620", "#F47A00", "#771155","#AA4488", "#CC99BB", "#114477", "#4477AA", "#77AADD", "#381C00", "#781156","#A51876","#D21E96","#E43FAD", "#117845","#18A55E","#1ED278","#3FE491","#6CEAAB")

#Duplicate phyloseq object to modify names and use for plot
Mothers_OTU_AFilt1_Preg_TaxPlot <- Mothers_OTU_AFilt1_Preg
#Changing sample names for plotting
sample_names(Mothers_OTU_AFilt1_Preg_TaxPlot) <- sample_data(Mothers_OTU_AFilt1_Preg_TaxPlot)$sample_id
taxa_names(Mothers_OTU_AFilt1_Preg_TaxPlot) <- paste((tax_table(Mothers_OTU_AFilt1_Preg_TaxPlot)[,8]), taxa_names(Mothers_OTU_AFilt1_Preg_TaxPlot), sep=".")
p <- plot_taxa(Mothers_OTU_AFilt1_Preg_TaxPlot, grouping_column = "T1Dstatus", method = "hellinger", number.taxa = 25, filename = NULL)
p + scale_fill_manual(values=Personal_colors) + scale_color_manual(values=Personal_colors) + theme(legend.title = element_text(size=15, face="bold"), legend.text = element_text(size=15, face="italic"), axis.text.y =element_text(size=15))  
```

#Alpha diversity 

Use GEEs to account for clustering due to repeated measures, and test for differences in alpha diversity between T1D and nonT1D. Include T1D x Days interaction term to test the possibility that the change in alpha diversity over time differs between T1D and nonT1D.

'Mean-centred' values are being used for gestational days, age and BMI were calculated. This ensures that the model coefficients presented are meaningful. The 'geepack' package defaults to using the empirical (robust or 'sandwich') estimator, which means that the estimates are robust to misspecification of the correlation structure.

```{r DivCalc, echo=FALSE}
##  Calculate observed richness  ###
#Obtain presence/absence counts to calculate number of observed OTUs (this is done because the original OTU table is normalized at relative abundance, decimal numbers)
DivCal_R <- estimate_richness(Mothers_OTU_AFilt1_Preg, measures=c("Observed"))
# make a data-frame with sample_data 
DivCal_R_df <- data.frame(DivCal_R, sample_data(Mothers_OTU_AFilt1_Preg))
DivCal_Sort <- arrange(DivCal_R_df, Days)
```

##Alpha diversity (T1D vs. non-T1D continuous time [Days])

```{r ATest, echo=FALSE}
# Mean-centre days, age at LMP and BMI (adjustment variables) so that the coefficients from the models are meaningful
meandays <- with(DivCal_R_df, mean(Days))
DivCal_R_df$days_c <- with(DivCal_R_df, Days - meandays)

meanage <- with(DivCal_R_df, mean(Age_LMP))
DivCal_R_df$age_c <- with(DivCal_R_df, Age_LMP - meanage)

meanbmi <- with(DivCal_R_df, mean(BMI_conception))
DivCal_R_df$bmi_c <- with(DivCal_R_df, BMI_conception - meanbmi)

DivCal_R_df$nullip <-(DivCal_R_df$Nulliparous)
DivCal_R_df$nullip <- factor(DivCal_R_df$nullip, levels=c(0,1), labels=c("No", "Yes"))

DivCal_R_df$HLA <-factor(DivCal_R_df$HLA.6DRML)

DivCal_R_df$Tri <-factor(DivCal_R_df$TriCorr)

DivCal_R_df$t1dfactor <- factor(DivCal_R_df$T1Dstatus, levels=c("no","yes"), labels=c("nonT1D", "T1D"))

ob_gee2 <- geeglm(Observed ~ t1dfactor*days_c + age_c + nullip + bmi_c + HLA, data=DivCal_R_df, id=motherid, corstr="exchangeable")
ResObs2 <- summary(ob_gee2)
ResObs2
```

##Alpha diversity (T1D vs. non-T1D categorical time [Trimesters])

```{r AlphaBoxplots, fig.height=4, fig.width=5, echo=FALSE}
ob_gee3 <- geeglm(Observed ~ t1dfactor*Tri + age_c + nullip + bmi_c + HLA, data=DivCal_R_df, id=motherid, corstr="exchangeable")
ResObs3 <- summary(ob_gee3)
ResObs3

T1Dcol <- c("blue", "red") #blue: non-T1D, red: T1D
Rich_R_OTU <- plot_richness(Mothers_OTU_AFilt1_Preg, x="T1Dstatus", scales = "fixed", measures=c("Observed"))
Rich_R_OTU_P <-  Rich_R_OTU + geom_boxplot(data=Rich_R_OTU$data, aes(color=T1Dstatus), alpha=0.1, size=1)  +  theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + labs(color="T1D status")  + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)
Rich_R_OTU_P
```

#**OTU taxonomic level**

##Microbial community composition analysis at **OTU** level (Beta diversity analysis)

**Hypothesis**

Hypothesis: the gut microbiome taxonomic composition (i.e. Beta diversity) during pregnancy differs between women with and without T1D. In order to test if this hypothesis holds true, a measurement of the distance between each pair of samples is calculated (i.e. Bray-Curtis) and a repeated measure aware permutational analysis of variance (i.e. RMA-PERMANOVA) test is applied. A P-value <0.05 is considered to be significant, meaning that our hypothesis cannot be rejected. Borderline P-values are also considered positive.

**Problems with the available R function for Beta diversity analysis**

The current available function for performing hypothesis testing of differences in the microbiome composition between groups of samples is called Adonis (i.e. which perform a PERMANOVA test) and is part of the Rpackage **vegan** (Oksanen, J. et al). The main problem with this function is that if the metadata of interest does not vary with time (e.g. disease status, sex, etc.), adonis does not calculate the corresponding P value correctly, as it permutes levels within a subject. This does not make sense for something like disease status or sex as permuting within-subject will produce the exact same distribution each time you permute. This a known drawback of the adonis function for repeated measures.  


In order to get the correct P-value for a time-invariant metadata, the permutation procedure has to be altered such that it permutes the subjects rather than levels within subjects, which apply to our current pregnancy dataset. 


Here I'm running a script written by **Jason Lloyd-Price** from Curtis Huttenhower lab in Harvard. This script is being used instead of the regular Adonis (PERMANOVA test) from the R package Vegan because as stated above, our metadata of interest (i.e. T1D status) does not vary with time, we  have a mixture of data which changes within and between an individual and we also have unequal group sizes (i.e. we do not always have exactly 3 samples per woman). For those three reasons, it was recommended to use his script to perform the RMA-PERMANOVA analysis.


**PARAMETERS:**


**permute_within:** This data frame has samples on rows and metadata on columns. This should only contain metadata that varies within a block (i.e. it's a single-column data frame with only time/Days).*Since, the other metadata within the block should be the same between repeated measurements and this will not hold when we introduce other factor i.e. sequencing run, BMI, parity, gestational age (i.e. Days or trimester) and conception age , those five factors should be placed here*


**blocks**: This should just be a vector giving the group of each sample (i.e. the motherid vector; motherid is strictly accounting for personID but not for pregnancy per se). 


**block_data**: This data frame contains per-block metadata, with one row per block (motherid). It should only contain metadata pertaining to the blocks (i.e. T1Dstatus). *motherid is not numeric, ensure that the row names match the factor names in the motherid vector (blocks)*.

**metadata_order**: This is needed if you want to specify a particular order that the model should be fit in. Metadata earlier in the list will be fit and residualized first, so these should be **features we are NOT interested in and want to control for**. 


**Results from PERMANOVA test with repeat measure-aware permutations - Controlling for Days/Trimester, sequencing run, conception age, BMI, parity and HLA type**

Here the blocking factor is taking into account each mother who might have given samples from different trimesters or two different pregnancies (i.e. motherid is the name of the factor). Therefore, the factor "motherid" takes into account two different pregnancies from the same mother as one SubjectID to adjust for repeated measurements

The interaction between T1D status and time was also included in order to test if the differences between T1D and non-T1D women in the same or changes throguhout pregnancy.

###Interaction between T1D status and Time

```{r Seed1, echo=FALSE}
set.seed(201)
```

```{r DistMatMetadata, echo=FALSE}
#Calculating bray-Curtis distace matrix
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg)))
```

```{r RMAP, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Days <- as.numeric(as.character(Meta_df$Days))
PW$Days <- Days
PW$T1D_Time_Interaction <- (Meta_df$T1Dstatus == "yes") * PW$Days ## To test the interaction as a continuous value aka Days!
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r RMAPeval, echo=FALSE}
#Calling RMA-PERMANOVA function
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub)
```

```{r ResultRMAP, echo=FALSE}
#P-value for interaction of T1D and time
RMAPres
RT1DTimeInt <- round(RMAPres$aov.tab$`Pr(>F)`[3], 4)
RT1DTimeInt
```

Due to the interaction between T1D and time having a significant P-value, differences in beta diversity between T1D and non-T1D was assessed by trimester using a normal PERMANOVA with the adonis function.

Note: The function relevel was used to change the trimester used as reference in the test (e.g. when testing for differences between T2 and T3).

###T1D status in trimester 1

```{r DistMatxT1, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg_T1, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T1)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
```

```{r RMAPxTri, echo=FALSE}
AdonisR <- adonis(D ~  SeqRun + Nulliparous + Age_LMP + BMI_conception + HLA.6DRML + T1Dstatus, data=Meta_dfTri)
AdonisR
```

```{r ResultRMAPAdonisT1D, echo=FALSE}
#P-value for T1D and time
RT1D <- round(AdonisR$aov.tab$`Pr(>F)`[6], 4)
RT1D
```

**Beta diversity Plot**

```{r PlotT1DxT1,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T1, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T1, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  + ggtitle( paste( "P-value =", RT1D ) )  
p_OTU12pr
```

###T1D status in trimester 2

```{r DistMatxT2, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg_T2, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T2)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r PlotT1DxT2,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T2, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T2, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  + ggtitle( paste( "P-value =", RT1D ) )  
p_OTU12pr
```

###T1D status in trimester 3

```{r DistMatxT3, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg_T3, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T3)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r PlotT1DxT3,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T3, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T3, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  + ggtitle( paste( "P-value =", RT1D ) )  
p_OTU12pr
```

###Days within T1D

```{r ChwT1D, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_PregwT1D, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregwT1D))
Meta_df <- as.data.frame(Meta)
```

```{r RMAPwT1D, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Days <- as.numeric(as.character(Meta_df$Days))
PW$Days <- Days

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r RMAPevaDays, echo=FALSE}
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub, 
    metadata_order = c(names(Meta_sub), names(PW)))
```

```{r ResultRMAPwT1D, echo=FALSE}
RMAPres
RDays <- round(RMAPres$aov.tab$`Pr(>F)`[6], 4)
```

**Beta diversity Plot**

```{r BCPCoADayswT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwT1D, method="PCoA", distance="bray")
mid<-mean(sample_data(Mothers_OTU_AFilt1_PregwT1D)$Days)
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="Days")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RDays ) ) + scale_color_gradient2(midpoint=161, low="blue", mid="white", high="red")
p_OTU12pr
```

###Trimesters within T1D

```{r RMAPTriwT1D, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ResultRMAP2wT1D, echo=FALSE}
RMAPres
RTri <- round(RMAPres$aov.tab$`Pr(>F)`[6], 4)
```

Note: The function relevel was used to change the trimester used as reference in the test (e.g. when testing for differences between T2 and T3).

**Beta diversity Plot**

```{r BCPCoATriwT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwT1D, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="TriCorr")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RTri ) )
p_OTU12pr
```

###Days within non-T1D

```{r ChDayswNT1D, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_PregwNT1D, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregwNT1D))
Meta_df <- as.data.frame(Meta)
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r BCPCoADayswNT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwNT1D, method="PCoA", distance="bray")
mid<-mean(sample_data(Mothers_OTU_AFilt1_PregwNT1D)$Days)
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwNT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="Days")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RDays ) ) + scale_color_gradient2(midpoint=161, low="blue", mid="white", high="red")
p_OTU12pr
```

###Trimesters within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r BCPCoATriwNT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwNT1D, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwNT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="TriCorr")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RTri ) )
p_OTU12pr
```
 
###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r Tri1OF, echo=FALSE}
TOF <- "T1" 
```

```{r Age, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Age_Time_Interaction <- (Meta_df$TriCorr == TOF) * PW$Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r RMAPevaOFactors, echo=FALSE}
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub, 
    metadata_order = c(names(Meta_sub), names(PW)))
RMAPres
```

#####Age at conception interaction with time (reference: trimester 2)

```{r Tri2OF, echo=FALSE}
TOF <- "T2" 
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r Tri3OF, echo=FALSE}
TOF <- "T3" 
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r AgeNI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r BMI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$BMI_Time_Interaction <- (Meta_df$TriCorr == TOF) * PW$BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r BMINI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r HLArefDR34, echo=FALSE}
HLAr <- "DR34"
```

```{r HLA, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$HLA_Time_Interaction <- (Meta_df$HLA.6DRML == HLAr) * PW$Days #FOR HLA DR34 DRXX Group3o4

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r HLArefDRXX, echo=FALSE}
HLAr <- "DRXX"
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r HLArefDR3o4, echo=FALSE}
HLAr <- "Group3o4"
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r HLANI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
T1Dstatus <- Meta_subj$T1Dstatus
Meta_sub <- as.data.frame(T1Dstatus)
rownames(Meta_sub) <- Meta_subj$motherid
HLA <- as.factor(Meta_subj$HLA.6DRML)
Meta_sub$HLA <- HLA
```

```{r RMAPevaOFactorsNI, echo=FALSE}
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub)
RMAPres
```

####Parity

#####Parity interaction with time

```{r Parity, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Parity_Time_Interaction <- (PW$Parity == "1") * PW$Days

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction)

```{r ParityNI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r MOD.CER, echo=FALSE}
MOD <- "CER"
```

```{r MOD, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$MOD_Time_Interaction <- (Meta_df$MOD == MOD) * PW$Days #FOR CER CE and LAB


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
MOD <- as.factor(Meta_subj$MOD)
Meta_sub$MOD <- MOD
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r MOD.CE, echo=FALSE}
MOD <- "CE"
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r MOD.Lab, echo=FALSE}
MOD <- "LAB"
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction) 

```{r MODNI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
MOD <- as.factor(Meta_subj$MOD)
Meta_sub$MOD <- MOD
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r Carbohydrate, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, carbohydrate_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$carbohydrate_g <- as.numeric(as.character(Meta_df$carbohydrate_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Carb_Time_Interaction <- (Meta_df$TriCorr == TOF) * Meta_df$carbohydrate_g

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Carbs <- as.numeric(as.character(Meta_subj$carbohydrate_g)) 
Meta_sub$Carbs <- Carbs
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r CarbohydrateNI, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, carbohydrate_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$carbohydrate_g <- as.numeric(as.character(Meta_df$carbohydrate_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Carbs <- as.numeric(as.character(Meta_subj$carbohydrate_g)) 
Meta_sub$Carbs <- Carbs
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r Fibre, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, fibre_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$fibre_g <- as.numeric(as.character(Meta_df$fibre_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Fiber_Time_Interaction <- (Meta_df$TriCorr == TOF) * Meta_df$fibre_g

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Fiber <- as.numeric(as.character(Meta_subj$fibre_g)) 
Meta_sub$Fiber <- Fiber
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r FibreNI, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, fibre_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$fibre_g <- as.numeric(as.character(Meta_df$fibre_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Fiber <- as.numeric(as.character(Meta_subj$fibre_g)) 
Meta_sub$Fiber <- Fiber
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r AG15, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregG <- subset_samples(Mothers_OTU_AFilt1_Preg, value_15ag != "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregG))
Meta_df <- as.data.frame(Meta)
D <- distance(Mothers_OTU_AFilt1_PregG, method="bray")
Meta_df$value_15ag <- as.numeric(as.character(Meta_df$value_15ag))

#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
AG15 <- as.numeric(as.character(Meta_df$value_15ag))
PW$AG15 <- AG15
PW$AG15_T1D_Interaction <- (Meta_df$T1Dstatus == "yes") * PW$AG15


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r T1DG, echo=FALSE}
T1DG <- "yes"
```

```{r AG15T1D, echo=FALSE}
Mothers_OTU_AFilt1_PregGT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%T1DG)
Mothers_OTU_AFilt1_PregG <- subset_samples(Mothers_OTU_AFilt1_PregGT1D, value_15ag != "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregG))
Meta_df <- as.data.frame(Meta)
D <- distance(Mothers_OTU_AFilt1_PregG, method="bray")
Meta_df$value_15ag <- as.numeric(as.character(Meta_df$value_15ag))

#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
AG15 <- as.numeric(as.character(Meta_df$value_15ag))
PW$AG15 <- AG15


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r NT1DG, echo=FALSE}
T1DG <- "no"
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG (no interaction)

```{r AG15NI, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregG <- subset_samples(Mothers_OTU_AFilt1_Preg, value_15ag != "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregG))
Meta_df <- as.data.frame(Meta)
D <- distance(Mothers_OTU_AFilt1_PregG, method="bray")
Meta_df$value_15ag <- as.numeric(as.character(Meta_df$value_15ag))

#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
AG15 <- as.numeric(as.character(Meta_df$value_15ag))
PW$AG15 <- AG15

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Pre-eclampsia (within T1D women)

```{r Preeclampsia, echo=FALSE}
Mothers_OTU_AFilt1_PregGT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))
Mothers_OTU_AFilt1_PregG <- subset_samples(Mothers_OTU_AFilt1_PregGT1D, Preeclampsia != "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregG))
Meta_df <- as.data.frame(Meta)
D <- distance(Mothers_OTU_AFilt1_PregG, method="bray")

#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Preeclampsia <- as.factor(Meta_df$Preeclampsia)
PW$Preeclampsia <- Preeclampsia
#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

##Differential abundance analysis Species taxonomic level

**Differential abundance analysis with limma using voom pipeline with structural zeros - Gordon**

Improved analysis using TMM normalization and voom with structural zeros.

A linear model is fit to the data and differential abundance is assessed using empirical bayes. The false discovery rate (FDR) for this analysis is set at 5%. For a OTU to be classified as differentially abundant (DA), its change in abundance between the groups (T1D vs nonT1D) must be significant. In the results tables to follow, the genes that are DE are those that have an adjusted p-value less than the FDR. Note that adjusted p-value is used opposed to the initial p-value as it has been adjusted for multiple testing. Therefore any adjusted p-value less than 0.05 is deemed statistically significant, identifying the associated gene as DA. 

Those genes that are DA are then determined to be more- or less abundant depending on the direction of their log-fold change. Those OTUs with a positive log-fold change are more abundant, while those OTUs with a negative log-fold change are less abundant. 

**Controlling for multiple measurements, sequencing run, conception age, BMI, parity and HLA type**

```{r de1, fig.height=7, fig.width=9, echo=FALSE}
Counts <- as.data.frame(otu_table(Mothers_OTU_AFilt1_Preg))
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg)))
Meta_df$SeqRun <- as.factor(Meta_df$SeqRun)
Meta_df$HLA.6DRML <- as.factor(Meta_df$HLA.6DRML)
Meta_df$Days <- as.integer(as.character(Meta_df$Days))
Meta_df$Age_LMP <- as.numeric(as.character(Meta_df$Age_LMP))
Meta_df$BMI_conception <- as.numeric(as.character(Meta_df$BMI_conception))
Meta_df$Nulliparous <- factor(Meta_df$Nulliparous, levels=c(0,1), labels=c("No", "Yes"))
```

```{r design, echo=FALSE}
#Using combined factor Trimester and T1Dstatus
design <- model.matrix(~0 + TriwT1D + SeqRun  + Nulliparous + Age_LMP + BMI_conception + HLA.6DRML, data=Meta_df)
```

```{r de2, fig.height=7, fig.width=9, echo=FALSE}
dge <- DGEList(Counts, group= Meta_df$T1Dstatus)
dgeTMM <- edgeR::calcNormFactors(dge, method = "TMM")
v_OTU <- voom(dgeTMM, design = design , plot = FALSE)

##Changes made by Gordon Smyth's suggestion
PoissonFit <- glmFit(dgeTMM,design,dispersion=0,prior.count=0)
StructuralZero <- (PoissonFit$fitted.values < 1e-8 & dgeTMM$counts < 1e-8)
v_OTU_NA <- v_OTU
v_OTU_NA$E[StructuralZero] <- NA
```

```{r de3, echo=FALSE}
## Contrast nonT1D_vs_T1D is composed of the addition of all trimesters in T1D and divided by 3 (average) vs the same for nonT1D to answer if there are differences in general independent of Trimester
cont <- makeContrasts(nonT1D_vs_T1D=(TriwT1DT1.no + TriwT1DT2.no + TriwT1DT3.no)/3 - (TriwT1DT1.yes + TriwT1DT2.yes + TriwT1DT3.yes)/3, T1= TriwT1DT1.no - TriwT1DT1.yes, T2= TriwT1DT2.no - TriwT1DT2.yes, T3= TriwT1DT3.no - TriwT1DT3.yes, T1_vs_T2=(TriwT1DT1.no + TriwT1DT1.yes)/2 - (TriwT1DT2.no + TriwT1DT2.yes)/2, T2_vs_T3=(TriwT1DT2.no + TriwT1DT2.yes)/2 - (TriwT1DT3.no + TriwT1DT3.yes)/2, T1_vs_T3=(TriwT1DT1.no + TriwT1DT1.yes)/2 - (TriwT1DT3.no + TriwT1DT3.yes)/2, T1DT1vsT2=TriwT1DT1.yes -TriwT1DT2.yes, T1DT2vsT3=TriwT1DT2.yes -TriwT1DT3.yes, T1DT1vsT3=TriwT1DT1.yes -TriwT1DT3.yes, noT1DT1vsT2=TriwT1DT1.no -TriwT1DT2.no, noT1DT2vsT3=TriwT1DT2.no -TriwT1DT3.no, noT1DT1vsT3=TriwT1DT1.no -TriwT1DT3.no, levels=design)

#Calculate correlation between repeated measurements (i.e. sample from the same woman)
corfit_NA <- duplicateCorrelation(v_OTU_NA, design, block = Meta_df$motherid)
fit_NA <- lmFit(v_OTU_NA, design = design, block = Meta_df$motherid, correlation = corfit_NA$consensus.correlation)
fit <- lmFit(v_OTU, design = design, block = Meta_df$motherid, correlation = corfit_NA$consensus.correlation)
#Substitute means and residuals calculated with structural zeroes
fit$sigma <- fit_NA$sigma
fit$df.residual <- fit_NA$df.residual
fit$Amean <- fit_NA$Amean
#Fit model with contrasts
fit <- contrasts.fit(fit, contrasts = cont)
fit <- eBayes(fit, robust=FALSE)
DT<- decideTests(fit)
summary(DT)
```

Differentially abundant species were found between T1D and non-T1D women across pregnancy and within each trimester. No differentially abundant species were detected between trimesters in samples form T1D and non-T1D women together or separatelly.

**Results for contrasts with significant differentially abundant species are shown below**

Only those taxa with prevalence > 50% and LogFC > 0.5 of < -0.5 are considered as biologically relevant in our analyses.

###Across all trimesters (T1D vs non-T1D)

```{r nonT1D_vs_T1D, echo=FALSE}
tt_nonT1D_vs_T1D <- topTable(fit, coef = "nonT1D_vs_T1D", n=Inf)
```

```{r nonT1D_vs_T1D_Subset, fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
#Only taxa with Padjusted < 0.1
DA_GG <- (row.names(tt_nonT1D_vs_T1D))[tt_nonT1D_vs_T1D$adj.P.Val<0.1]
```

```{r Counts, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
# Relative abundance counts
Mothers_OTU_AFilt1_Preg_Rel <- transform_sample_counts(Mothers_OTU_AFilt1_Preg_RA, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Mothers_OTU_AFilt1_Preg_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Mothers_OTU_AFilt1_Preg_Rel)$T1Dstatus)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Mothers_OTU_AFilt1_Preg_Rel)$T1Dstatus)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_nonT1D_vs_T1D[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "nonT1D:mean%", "Prev%", "T1D:mean%", "T1D:Prev%")
TableCom$Feature <- rownames(TableCom)
TableCom2 <- filter(TableCom, `Prev%` >= !!50 | `T1D:Prev%` >= !!50)
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])

TableCom <- cbind(NameD, tt_nonT1D_vs_T1D[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50)
TableCom2
}
} else {
  print("No DA taxa")
}
```

From 11 OTUs that limma identified as differentially abundant, only one (further classified as Bacteroides caccae in blast), had >50% prevalence in either of the groups (i.e. T1D or non-T1D). Here, we will just show hoe to plot mean and standard error of this differentially abundant taxa, however, this code can be used to plot any taxa that was differentially abundant in the rest of the analysis.

```{r ErrorBarsOTUAll, fig.height=5, fig.width=8, echo=FALSE}
T1Dcol <- c("blue", "red")
namTax <- c("Bacteroides caccae")
DA_GG <- as.character(TableCom2[1,9])
LT <- length(DA_GG)
for (i in 1:LT) {
DA1 <- DA_GG[i]
Mothers_OTU_AFilt1_Preg_RA_DA_GG <- prune_taxa(DA1, Mothers_OTU_AFilt1_Preg_Rel)
Sample <- sample_data(Mothers_OTU_AFilt1_Preg_RA_DA_GG)[,c("TriCorr", "T1Dstatus")]
Order <- data.frame(t(otu_table(Mothers_OTU_AFilt1_Preg_RA_DA_GG)))
colnames(Order) <- "TaxaCounts"
TablesB <- cbind(Sample, Order)

tgc <- summarySE(TablesB, measurevar="TaxaCounts", groupvars=c("T1Dstatus","TriCorr"))
pd <- position_dodge(0.1)

p2 <- ggplot(tgc, aes(x=TriCorr, y=TaxaCounts, colour=T1Dstatus, group=T1Dstatus))
nam <- paste("P",1:LT,sep="")
assign(nam[i],  p2 + geom_errorbar(aes(ymin=TaxaCounts-se, ymax=TaxaCounts+se), width=.1, position=pd) + geom_line(position=pd, size=1.5) + geom_point(position=pd, size=3, shape=21, fill="white") + labs(x = "Trimesters", y="Rel. abundance (%)")  + theme(axis.title.x = element_blank(), axis.title.y = element_blank()) + theme(axis.text.y =element_text(size=14), axis.text.x =element_blank()) + theme(legend.title = element_text(size=15, face="bold"))  + theme(legend.text = element_text(size = 15))  + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol) + theme(legend.position='none') + ggtitle(paste(namTax[i])) + scale_y_continuous(labels = scales::number_format(accuracy = 0.1)))
}
print(P1)
```

### Only Trimester 1 (nonT1D vs T1D)

```{r T1_nonT1D_vs_T1D, echo=FALSE}
tt_T1 <- topTable(fit, coef = "T1", n=Inf)
```

```{r T1_nonT1D_vs_T1D_Subset, echo=FALSE}
DA_GG <- (row.names(tt_T1))[tt_T1$adj.P.Val<0.1]
```

```{r CountsT1, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
Dummy1 <- Mothers_OTU_AFilt1_Preg_T1
# Relative abundance counts
Dummy1_Rel <- transform_sample_counts(Dummy1, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy1_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy1_Rel)$T1Dstatus)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy1_Rel)$T1Dstatus)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
rownames(TableCom) <- c(1:NU)
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "nonT1D:mean%", "Prev%", "T1D:mean%", "T1D:Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50 | `T1D:Prev%` >= !!50)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])

TableCom <- cbind(NameD, tt_T1[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### Only Trimester 2 (nonT1D vs T1D)

```{r T2_nonT1D_vs_T1D, echo=FALSE}
tt_T2 <- topTable(fit, coef = "T2", n=Inf)
```

```{r T2_nonT1D_vs_T1D_Subset, fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
DA_GG <- (row.names(tt_T2))[tt_T2$adj.P.Val<0.1]
```

```{r CountsT2, echo=FALSE}
# Relative abundance counts
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
Dummy1 <- Mothers_OTU_AFilt1_Preg_T2
# Relative abundance counts
Dummy1_Rel <- transform_sample_counts(Dummy1, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy1_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy1_Rel)$T1Dstatus)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy1_Rel)$T1Dstatus)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
rownames(TableCom) <- c(1:NU)
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "nonT1D:mean%", "Prev%", "T1D:mean%", "T1D:Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50 | `T1D:Prev%` >= !!50)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])

TableCom <- cbind(NameD, tt_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### Only Trimester 3 (nonT1D vs T1D)

```{r T3_nonT1D_vs_T1D, echo=FALSE}
tt_T3 <- topTable(fit, coef = "T3", n=Inf)
```

```{r T3_nonT1D_vs_T1D_Subset, fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
DA_GG <- (row.names(tt_T3))[tt_T3$adj.P.Val<0.1]
```

```{r CountsT3, echo=FALSE}
# Relative abundance counts
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
Dummy1 <- Mothers_OTU_AFilt1_Preg_T3
# Relative abundance counts
Dummy1_Rel <- transform_sample_counts(Dummy1, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy1_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy1_Rel)$T1Dstatus)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy1_Rel)$T1Dstatus)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
rownames(TableCom) <- c(1:NU)
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "nonT1D:mean%", "Prev%", "T1D:mean%", "T1D:Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50 | `T1D:Prev%` >= !!50)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])

TableCom <- cbind(NameD, tt_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1,4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification",  "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
TableCom2 <- filter(TableCom, `Prev%` >= !!50)
TableCom2
 
}
} else {
  print("No DA taxa")
}
```

### T1 vs T2 (T1D and non-T1D data)

```{r T1_vs_T2, echo=FALSE}
tt_T1_vs_T2 <- topTable(fit, coef = "T1_vs_T2", n=Inf)
```

```{r T1_vs_T2_Subset, echo=FALSE}
DA_GG <- (row.names(tt_T1_vs_T2))[tt_T1_vs_T2$adj.P.Val<0.1]
```

```{r CountsTriCorr12, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
# Relative abundance counts
Mothers_OTU_AFilt1_Preg_Rel <- transform_sample_counts(Mothers_OTU_AFilt1_Preg, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Mothers_OTU_AFilt1_Preg_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1_vs_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3), round(x_RelFil_Res[,3],3), round(x_PA_RelFil_Res[,3],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T2:mean%", "T2Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T1Prev%` >= !!50 | `T2Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1_vs_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T2", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### T2 vs T3 (T1D and non-T1D data)

```{r T2_vs_T3, echo=FALSE}
tt_T2_vs_T3 <- topTable(fit, coef = "T2_vs_T3", n=Inf)
```

```{r T2_vs_T3_Subset, fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
DA_GG <- (row.names(tt_T2_vs_T3))[tt_T2_vs_T3$adj.P.Val<0.1]
```

```{r CountsTriCorr23, echo=FALSE}
# Relative abundance counts
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
# Relative abundance counts
Mothers_OTU_AFilt1_Preg_Rel <- transform_sample_counts(Mothers_OTU_AFilt1_Preg, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Mothers_OTU_AFilt1_Preg_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T2_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3), round(x_RelFil_Res[,3],3), round(x_PA_RelFil_Res[,3],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T2:mean%", "T2Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T2Prev%` >= !!50 | `T3Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T2_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T2", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### T1 vs T3 (T1D and non-T1D data)

```{r T1_vs_T3, echo=FALSE}
tt_T1_vs_T3 <- topTable(fit, coef = "T1_vs_T3", n=Inf)
```

```{r T1_vs_T3_Subset, fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
DA_GG <- (row.names(tt_T1_vs_T3))[tt_T1_vs_T3$adj.P.Val<0.1]
```

```{r CountsTriCorr13, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg)
# Relative abundance counts
Mothers_OTU_AFilt1_Preg_Rel <- transform_sample_counts(Mothers_OTU_AFilt1_Preg, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Mothers_OTU_AFilt1_Preg_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3), round(x_RelFil_Res[,3],3), round(x_PA_RelFil_Res[,3],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T2:mean%", "T2Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T1Prev%` >= !!50 | `T3Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
} else {
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T2", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### T1 vs T2 (in women with T1D)

```{r T1DT1vsT2, echo=FALSE}
tt_T1DT1_vs_T2 <- topTable(fit, coef = "T1DT1vsT2", n=Inf)
```

```{r T1DT1vsT2_PH1, echo=FALSE}
DA_GG <- (row.names(tt_T1DT1_vs_T2))[tt_T1DT1_vs_T2$adj.P.Val<0.1]
```

```{r CountsT1DT12, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg_T1DT1T2)
Dummy2 <- Mothers_OTU_AFilt1_Preg_T1DT1T2
# Relative abundance counts
Dummy2_Rel <- transform_sample_counts(Dummy2, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy2_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy2_Rel)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy2_Rel)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1DT1_vs_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T2:mean%", "T2Prev%")
TableCom2 <- filter(TableCom, `T1Prev%` >= !!50 | `T2Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1DT1_vs_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T2")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}

```

### T2 vs T3 (in women with T1D)

```{r T1DT2vsT3, echo=FALSE}
tt_T1DT2_vs_T3 <- topTable(fit, coef = "T1DT2vsT3", n=Inf)
```

```{r T1DT2vsT3_PH1, echo=FALSE}
DA_GG <- (row.names(tt_T1DT2_vs_T3))[tt_T1DT2_vs_T3$adj.P.Val<0.1]
```

```{r CountsT1DT23, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg_T1DT2T3)
Dummy2 <- Mothers_OTU_AFilt1_Preg_T1DT2T3
# Relative abundance counts
Dummy2_Rel <- transform_sample_counts(Dummy2, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy2_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy2_Rel)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy2_Rel)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1DT2_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T2:mean%", "T2Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T2Prev%` >= !!50 | `T3Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1DT2_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T2", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### T1 vs T3 (in women with T1D)

```{r T1DT1vsT3, echo=FALSE}
tt_T1DT1_vs_T3 <- topTable(fit, coef = "T1DT1vsT3", n=Inf)
```

```{r T1DT1vsT3_PH1, echo=FALSE}
DA_GG <- (row.names(tt_T1DT1_vs_T3))[tt_T1DT1_vs_T3$adj.P.Val<0.1]
```

```{r CountsT1DT13, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg_T1DT1T3)
Dummy2 <- Mothers_OTU_AFilt1_Preg_T1DT1T3
# Relative abundance counts
Dummy2_Rel <- transform_sample_counts(Dummy2, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy2_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy2_Rel)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy2_Rel)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1DT1_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T1Prev%` >= !!50 | `T3Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_T1DT1_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### T1 vs T2 (in women without T1D)

```{r noT1DT1vsT2, echo=FALSE}
tt_noT1DT1_vs_T2 <- topTable(fit, coef = "noT1DT1vsT2", n=Inf)
```

```{r noT1DT1vsT2_PH1, echo=FALSE}
DA_GG <- (row.names(tt_noT1DT1_vs_T2))[tt_noT1DT1_vs_T2$adj.P.Val<0.1]
```

```{r CountsnonT1DT12, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg_nonT1DT1T2)
Dummy2 <- Mothers_OTU_AFilt1_Preg_nonT1DT1T2
# Relative abundance counts
Dummy2_Rel <- transform_sample_counts(Dummy2, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy2_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy2_Rel)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy2_Rel)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_noT1DT1_vs_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T2:mean%", "T2Prev%")
TableCom2 <- filter(TableCom, `T1Prev%` >= !!50 | `T2Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_noT1DT1_vs_T2[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T2")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}

```

### T2 vs T3 (in women without T1D)

```{r noT1DT2vsT3, echo=FALSE}
tt_noT1DT2_vs_T3 <- topTable(fit, coef = "noT1DT2vsT3", n=Inf)
```

```{r noT1DT2vsT3_PH1, echo=FALSE}
DA_GG <- (row.names(tt_noT1DT2_vs_T3))[tt_noT1DT2_vs_T3$adj.P.Val<0.1]
```

```{r CountsnonT1DT23, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg_nonT1DT2T3)
Dummy2 <- Mothers_OTU_AFilt1_Preg_nonT1DT2T3
# Relative abundance counts
Dummy2_Rel <- transform_sample_counts(Dummy2, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy2_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy2_Rel)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy2_Rel)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_noT1DT2_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T2:mean%", "T2Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T2Prev%` >= !!50 | `T3Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_noT1DT2_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T2", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

### T1 vs T3 (in women without T1D)

```{r noT1DT1vsT3, echo=FALSE}
tt_noT1DT1_vs_T3 <- topTable(fit, coef = "noT1DT1vsT3", n=Inf)
```

```{r noT1DT1vsT3_PH1, echo=FALSE}
DA_GG <- (row.names(tt_noT1DT1_vs_T3))[tt_noT1DT1_vs_T3$adj.P.Val<0.1]
```

```{r CountsnonT1DT13, echo=FALSE}
if(length(DA_GG)>0){
Mothers_OTU_AFilt1_Preg_DA_GG = prune_taxa(DA_GG, Mothers_OTU_AFilt1_Preg_nonT1DT1T3)
Dummy2 <- Mothers_OTU_AFilt1_Preg_nonT1DT1T3
# Relative abundance counts
Dummy2_Rel <- transform_sample_counts(Dummy2, function(x) 100 * x/sum(x))
x_Rel <- as.data.frame(t(as.data.frame(otu_table(Dummy2_Rel))))
### Presence / absence
x_PA <- x_Rel
x_PA[x_PA >0] <- 1
## Average relative abundance (%)
x_RelFil<-split(x_Rel, sample_data(Dummy2_Rel)$TriCorr)
### Proportion present 
xFil_PA<-split(x_PA, sample_data(Dummy2_Rel)$TriCorr)

if(length(DA_GG)>1){
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){colMeans(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_noT1DT1_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res[,1],3), round(x_PA_RelFil_Res[,1],3), round(x_RelFil_Res[,2],3), round(x_PA_RelFil_Res[,2],3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "T1:mean%", "T1Prev%", "T3:mean%", "T3Prev%")
TableCom2 <- filter(TableCom, `T1Prev%` >= !!50 | `T3Prev%` >= !!50)
TableCom2 <- filter(TableCom2, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2 <- TableCom2[order(TableCom2$LogFC),]
TableCom2
} else {
## T1D vs nonT1D Average relative abundance (%)
x_RelFil_Res <-sapply(x_RelFil, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])}) 
x_PA_RelFil_Res <-sapply(xFil_PA, function(x){mean(x[, rownames(otu_table(Mothers_OTU_AFilt1_Preg_DA_GG))])*100})

x_RelFil_Res <- as.data.frame(x_RelFil_Res)
x_PA_RelFil_Res <- as.data.frame(x_PA_RelFil_Res)
NameD <- as.data.frame(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)[,TaxLv])
NU <- nrow(NameD)

TableCom <- cbind(NameD, tt_noT1DT1_vs_T3[rownames(tax_table(Mothers_OTU_AFilt1_Preg_DA_GG)),c(1, 4:5)], round(x_RelFil_Res,3), round(x_PA_RelFil_Res,3))
colnames(TableCom) <- c("Classification", "LogFC", "P.Val", "adj.P.Val", "mean%", "Prev%")
rownames(TableCom) <- c("T1", "T3")
TableCom2 <- filter(TableCom, LogFC >= !!0.5 | LogFC <= !!-0.5)
TableCom2
}
} else {
  print("No DA taxa")
}
```

#**Genus taxonomic level**

##Microbial community composition analysis at **Genus** level (Beta diversity analysis)

```{r Genus, echo=FALSE}
Mothers_OTU_AFilt1_Preg_TL <- tax_glom(Original_OTU, taxrank = "Genus")
taxa_names(Mothers_OTU_AFilt1_Preg_TL)<-tax_table(Mothers_OTU_AFilt1_Preg_TL)[,"Genus"]
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_TL

# Subset to have only mothers trimester 1 
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T1"))
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg_T1, TakeOne%in%c("Y"))
SSumT1 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T1))
Mothers_OTU_AFilt1_Preg_T1_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T1, sample.size = SSumT1, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T1 <- Mothers_OTU_AFilt1_Preg_T1_RA

# Subset to have only mothers trimester 2 
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg_T2, TakeOne%in%c("Y"))
SSumT2 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T2))
Mothers_OTU_AFilt1_Preg_T2_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T2, sample.size = SSumT2, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T2 <- Mothers_OTU_AFilt1_Preg_T2_RA

# Subset to have only mothers trimester 3 
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg_T3, TakeOne%in%c("Y"))
SSumT3 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T3))
Mothers_OTU_AFilt1_Preg_T3_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T3, sample.size = SSumT3, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T3 <- Mothers_OTU_AFilt1_Preg_T3_RA

#Rarefy main dataset
SSum <- min(sample_sums(Mothers_OTU_AFilt1_Preg))
Mothers_OTU_AFilt1_Preg_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg, sample.size = SSum, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_RA

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))

#Subset to have only mothers without T1D
Mothers_OTU_AFilt1_PregwNT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("no"))

Mothers_OTU_AFilt1_Preg_T1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T2.yes"))
Mothers_OTU_AFilt1_Preg_T1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_T1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T2.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.no", "T3.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T3.no"))

TaxLv<- as.integer("6")
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatxT2', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatxT3', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction)

```{r ref.label='ParityNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction) 

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction within T1)

```{r T1NI, echo=FALSE}
TNI <- "T1"
```

```{r CarbohydrateNITri, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregT1Carb <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%TNI)
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_PregT1Carb, carbohydrate_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$carbohydrate_g <- as.numeric(as.character(Meta_df$carbohydrate_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Carbs <- as.numeric(as.character(Meta_subj$carbohydrate_g)) 
Meta_sub$Carbs <- Carbs
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T2)

```{r T2NI, echo=FALSE}
TNI <- "T2"
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T3)

```{r T3NI, echo=FALSE}
TNI <- "T3"
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r FiberNITri, echo=FALSE}
##Filtering out sample with no information about fiber intake
Mothers_OTU_AFilt1_PregT1Fib <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%TNI)
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_PregT1Fib, fibre_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$fibre_g <- as.numeric(as.character(Meta_df$fibre_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Fiber <- as.numeric(as.character(Meta_subj$fibre_g)) 
Meta_sub$Fiber <- Fiber
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG (no interaction)

```{r ref.label='AG15NI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Pre-eclampsia (within T1D women)

```{r ref.label='Preeclampsia', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

##Differential abundance analysis Genus taxonomic level

**Controlling for multiple measurements, sequencing run, conception age, BMI, parity and HLA type**

```{r ref.label='de1', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='design', echo=FALSE}
```

```{r ref.label='de2', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='de3', echo=FALSE}
```

**Results for contrasts with significant differentially abundant strains shown below**

###Across all trimesters (T1D vs non-T1D)

```{r ref.label='nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='Counts', echo=FALSE}
```

### Only Trimester 1 (nonT1D vs T1D)

```{r ref.label='T1_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T1_nonT1D_vs_T1D_Subset', echo=FALSE}
```

```{r ref.label='CountsT1', echo=FALSE}
```

### Only Trimester 2 (nonT1D vs T1D)

```{r ref.label='T2_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T2_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CPlotT2', echo=FALSE}
```

```{r ref.label='CountsT2', echo=FALSE}
```

### Only Trimester 3 (nonT1D vs T1D)

```{r ref.label='T3_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T3_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsT3', echo=FALSE}
```

### T1 vs T2 (T1D and non-T1D data)

```{r ref.label='T1_vs_T2', echo=FALSE}
```

```{r ref.label='T1_vs_T2_Subset', echo=FALSE}
```

```{r ref.label='CountsTriCorr12', echo=FALSE}
```

### T2 vs T3 (T1D and non-T1D data)

```{r ref.label='T2_vs_T3', echo=FALSE}
```

```{r ref.label='T2_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr23', echo=FALSE}
```

Nothing left after filtering by prevalence and LogFC

### T1 vs T3 (T1D and non-T1D data)

```{r ref.label='T1_vs_T3', echo=FALSE}
```

```{r ref.label='T1_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr13', echo=FALSE}
```

### T1 vs T2 (in women with T1D)

```{r ref.label='T1DT1vsT2', echo=FALSE}
```

```{r ref.label='T1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT12', echo=FALSE}
```

### T2 vs T3 (in women with T1D)

```{r ref.label='T1DT2vsT3', echo=FALSE}
```

```{r ref.label='T1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT23', echo=FALSE}
```

### T1 vs T3 (in women with T1D)

```{r ref.label='T1DT1vsT3', echo=FALSE}
```

```{r ref.label='T1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT13', echo=FALSE}
```

### T1 vs T2 (in women without T1D)

```{r ref.label='noT1DT1vsT2', echo=FALSE}
```

```{r ref.label='noT1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT12', echo=FALSE}
```

### T2 vs T3 (in women without T1D)

```{r ref.label='noT1DT2vsT3', echo=FALSE}
```

```{r ref.label='noT1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT23', echo=FALSE}
```

### T1 vs T3 (in women without T1D)

```{r ref.label='noT1DT1vsT3', echo=FALSE}
```

```{r ref.label='noT1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT13', echo=FALSE}
```

#**Family taxonomic level**

##Microbial community composition analysis at **Family** level (Beta diversity analysis)


```{r Family, echo=FALSE}
Mothers_OTU_AFilt1_Preg_TL <- tax_glom(Original_OTU, taxrank = "Family")
taxa_names(Mothers_OTU_AFilt1_Preg_TL)<-tax_table(Mothers_OTU_AFilt1_Preg_TL)[,"Family"]
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_TL

# Subset to have only mothers trimester 1 
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T1"))
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg_T1, TakeOne%in%c("Y"))
SSumT1 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T1))
Mothers_OTU_AFilt1_Preg_T1_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T1, sample.size = SSumT1, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T1 <- Mothers_OTU_AFilt1_Preg_T1_RA

# Subset to have only mothers trimester 2 
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg_T2, TakeOne%in%c("Y"))
SSumT2 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T2))
Mothers_OTU_AFilt1_Preg_T2_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T2, sample.size = SSumT2, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T2 <- Mothers_OTU_AFilt1_Preg_T2_RA

# Subset to have only mothers trimester 3 
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg_T3, TakeOne%in%c("Y"))
SSumT3 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T3))
Mothers_OTU_AFilt1_Preg_T3_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T3, sample.size = SSumT3, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T3 <- Mothers_OTU_AFilt1_Preg_T3_RA

#Rarefy main dataset
SSum <- min(sample_sums(Mothers_OTU_AFilt1_Preg))
Mothers_OTU_AFilt1_Preg_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg, sample.size = SSum, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_RA

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))

#Subset to have only mothers without T1D
Mothers_OTU_AFilt1_PregwNT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("no"))

Mothers_OTU_AFilt1_Preg_T1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T2.yes"))
Mothers_OTU_AFilt1_Preg_T1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_T1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T2.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.no", "T3.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T3.no"))

TaxLv<- as.integer("5")
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatxT2', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatxT3', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ParityNITri, echo=FALSE}
Mothers_OTU_AFilt1_PregT1Par <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%TNI)
D <- distance(Mothers_OTU_AFilt1_PregT1Par, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregT1Par))
Meta_df <- as.data.frame(Meta)
#permute_within

seqRun <- as.numeric(as.character(Meta_df$SeqRun))
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction) 

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG (no interaction)

```{r ref.label='AG15NI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Pre-eclampsia (within T1D women)

```{r ref.label='Preeclampsia', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

##Differential abundance analysis Family taxonomic level

**Controlling for multiple measurements, sequencing run, conception age, BMI, parity and HLA type**

```{r ref.label='de1', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='design', echo=FALSE}
```

```{r ref.label='de2', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='de3', echo=FALSE}
```

**Results for contrasts with significant (or borderline significant) differences shown below**

###Across all trimesters (T1D vs non-T1D)

```{r ref.label='nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='Counts', echo=FALSE}
```

### Only Trimester 1 (nonT1D vs T1D)

```{r ref.label='T1_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T1_nonT1D_vs_T1D_Subset', echo=FALSE}
```

```{r ref.label='CountsT1', echo=FALSE}
```

### Only Trimester 2 (nonT1D vs T1D)

```{r ref.label='T2_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T2_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CPlotT2', echo=FALSE}
```

```{r ref.label='CountsT2', echo=FALSE}
```

### Only Trimester 3 (nonT1D vs T1D)

```{r ref.label='T3_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T3_nonT1D_vs_T1D_Subset', echo=FALSE}
```

```{r ref.label='CountsT3', echo=FALSE}
```

### T1 vs T2 (T1D and non-T1D data)

```{r ref.label='T1_vs_T2', echo=FALSE}
```

```{r ref.label='T1_vs_T2_Subset', echo=FALSE}
```

```{r ref.label='CountsTriCorr12', echo=FALSE}
```

### T2 vs T3 (T1D and non-T1D data)

```{r ref.label='T2_vs_T3', echo=FALSE}
```

```{r ref.label='T2_vs_T3_Subset', echo=FALSE}
```

```{r ref.label='CountsTriCorr23', echo=FALSE}
```

Nothing left after filtering by prevalence and LogFC

### T1 vs T3 (T1D and non-T1D data)

```{r ref.label='T1_vs_T3', echo=FALSE}
```

```{r ref.label='T1_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr13', echo=FALSE}
```

### T1 vs T2 (in women with T1D)

```{r ref.label='T1DT1vsT2', echo=FALSE}
```

```{r ref.label='T1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT12', echo=FALSE}
```

### T2 vs T3 (in women with T1D)

```{r ref.label='T1DT2vsT3', echo=FALSE}
```

```{r ref.label='T1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT23', echo=FALSE}
```

### T1 vs T3 (in women with T1D)

```{r ref.label='T1DT1vsT3', echo=FALSE}
```

```{r ref.label='T1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT13', echo=FALSE}
```

### T1 vs T2 (in women without T1D)

```{r ref.label='noT1DT1vsT2', echo=FALSE}
```

```{r ref.label='noT1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT12', echo=FALSE}
```

### T2 vs T3 (in women without T1D)

```{r ref.label='noT1DT2vsT3', echo=FALSE}
```

```{r ref.label='noT1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT23', echo=FALSE}
```

### T1 vs T3 (in women without T1D)

```{r ref.label='noT1DT1vsT3', echo=FALSE}
```

```{r ref.label='noT1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT13', echo=FALSE}
```

#**Order taxonomic level**

##Microbial community composition analysis at **Order** level (Beta diversity analysis)


```{r Order, echo=FALSE}
Mothers_OTU_AFilt1_Preg_TL <- tax_glom(Original_OTU, taxrank = "Order")
taxa_names(Mothers_OTU_AFilt1_Preg_TL)<-tax_table(Mothers_OTU_AFilt1_Preg_TL)[,"Order"]
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_TL

# Subset to have only mothers trimester 1 
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T1"))
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg_T1, TakeOne%in%c("Y"))
SSumT1 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T1))
Mothers_OTU_AFilt1_Preg_T1_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T1, sample.size = SSumT1, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T1 <- Mothers_OTU_AFilt1_Preg_T1_RA

# Subset to have only mothers trimester 2 
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg_T2, TakeOne%in%c("Y"))
SSumT2 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T2))
Mothers_OTU_AFilt1_Preg_T2_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T2, sample.size = SSumT2, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T2 <- Mothers_OTU_AFilt1_Preg_T2_RA

# Subset to have only mothers trimester 3 
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg_T3, TakeOne%in%c("Y"))
SSumT3 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T3))
Mothers_OTU_AFilt1_Preg_T3_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T3, sample.size = SSumT3, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T3 <- Mothers_OTU_AFilt1_Preg_T3_RA

#Rarefy main dataset
SSum <- min(sample_sums(Mothers_OTU_AFilt1_Preg))
Mothers_OTU_AFilt1_Preg_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg, sample.size = SSum, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_RA

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))

#Subset to have only mothers without T1D
Mothers_OTU_AFilt1_PregwNT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("no"))

Mothers_OTU_AFilt1_Preg_T1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T2.yes"))
Mothers_OTU_AFilt1_Preg_T1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_T1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T2.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.no", "T3.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T3.no"))

TaxLv<- as.integer("3")
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction) 

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG (no interaction)

```{r ref.label='AG15NI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Pre-eclampsia (within T1D women)

```{r ref.label='Preeclampsia', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

##Differential abundance analysis Order taxonomic level

**Controlling for multiple measurements, sequencing run, conception age, BMI, parity and HLA type**

```{r ref.label='de1', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='design', echo=FALSE}
```

```{r ref.label='de2', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='de3', echo=FALSE}
```

###Across all trimesters (T1D vs non-T1D)

```{r ref.label='nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='Counts', echo=FALSE}
```

### Only Trimester 1 (nonT1D vs T1D)

```{r ref.label='T1_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T1_nonT1D_vs_T1D_Subset', echo=FALSE}
```

```{r ref.label='CountsT1', echo=FALSE}
```

### Only Trimester 2 (nonT1D vs T1D)

```{r ref.label='T2_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T2_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CPlotT2', echo=FALSE}
```

```{r ref.label='CountsT2', echo=FALSE}
```

After filtering low prevalent taxa, no species was significantly different between T1D and non-T1D women in this comparison. 

### Only Trimester 3 (nonT1D vs T1D)

```{r ref.label='T3_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T3_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsT3', echo=FALSE}
```

### T1 vs T2 (T1D and non-T1D data)

```{r ref.label='T1_vs_T2', echo=FALSE}
```

```{r ref.label='T1_vs_T2_Subset', echo=FALSE}
```

```{r ref.label='CountsTriCorr12', echo=FALSE}
```

### T2 vs T3 (T1D and non-T1D data)

```{r ref.label='T2_vs_T3', echo=FALSE}
```

```{r ref.label='T2_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr23', echo=FALSE}
```

Nothing left after filtering by prevalence and LogFC

### T1 vs T3 (T1D and non-T1D data)

```{r ref.label='T1_vs_T3', echo=FALSE}
```

```{r ref.label='T1_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr13', echo=FALSE}
```

### T1 vs T2 (in women with T1D)

```{r ref.label='T1DT1vsT2', echo=FALSE}
```

```{r ref.label='T1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT12', echo=FALSE}
```

### T2 vs T3 (in women with T1D)

```{r ref.label='T1DT2vsT3', echo=FALSE}
```

```{r ref.label='T1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT23', echo=FALSE}
```

### T1 vs T3 (in women with T1D)

```{r ref.label='T1DT1vsT3', echo=FALSE}
```

```{r ref.label='T1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT13', echo=FALSE}
```

### T1 vs T2 (in women without T1D)

```{r ref.label='noT1DT1vsT2', echo=FALSE}
```

```{r ref.label='noT1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT12', echo=FALSE}
```

### T2 vs T3 (in women without T1D)

```{r ref.label='noT1DT2vsT3', echo=FALSE}
```

```{r ref.label='noT1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT23', echo=FALSE}
```

### T1 vs T3 (in women without T1D)

```{r ref.label='noT1DT1vsT3', echo=FALSE}
```

```{r ref.label='noT1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT13', echo=FALSE}
```

#**Phylum taxonomic level**

##Microbial community composition analysis at **Phylum** level (Beta diversity analysis)

```{r Phylum, echo=FALSE}
Mothers_OTU_AFilt1_Preg_TL <- tax_glom(Original_OTU, taxrank = "Phylum")
taxa_names(Mothers_OTU_AFilt1_Preg_TL)<-tax_table(Mothers_OTU_AFilt1_Preg_TL)[,"Phylum"]
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_TL

# Subset to have only mothers trimester 1 
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T1"))
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg_T1, TakeOne%in%c("Y"))
SSumT1 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T1))
Mothers_OTU_AFilt1_Preg_T1_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T1, sample.size = SSumT1, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T1 <- Mothers_OTU_AFilt1_Preg_T1_RA

# Subset to have only mothers trimester 2 
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg_T2, TakeOne%in%c("Y"))
SSumT2 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T2))
Mothers_OTU_AFilt1_Preg_T2_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T2, sample.size = SSumT2, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T2 <- Mothers_OTU_AFilt1_Preg_T2_RA

# Subset to have only mothers trimester 3 
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg_T3, TakeOne%in%c("Y"))
SSumT3 <- min(sample_sums(Mothers_OTU_AFilt1_Preg_T3))
Mothers_OTU_AFilt1_Preg_T3_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg_T3, sample.size = SSumT3, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg_T3 <- Mothers_OTU_AFilt1_Preg_T3_RA

#Rarefy main dataset
SSum <- min(sample_sums(Mothers_OTU_AFilt1_Preg))
Mothers_OTU_AFilt1_Preg_RA <- rarefy_even_depth(Mothers_OTU_AFilt1_Preg, sample.size = SSum, rngseed = 711, replace=FALSE, trimOTUs = TRUE)
Mothers_OTU_AFilt1_Preg <- Mothers_OTU_AFilt1_Preg_RA

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))

#Subset to have only mothers without T1D
Mothers_OTU_AFilt1_PregwNT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("no"))

Mothers_OTU_AFilt1_Preg_T1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T2.yes"))
Mothers_OTU_AFilt1_Preg_T1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_T1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.yes", "T3.yes"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T2.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT2T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T2.no", "T3.no"))
Mothers_OTU_AFilt1_Preg_nonT1DT1T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriwT1D%in%c("T1.no", "T3.no"))

TaxLv<- as.integer("2")
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction) 

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Carbohydrate (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='CarbohydrateNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Fibre (no interaction within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='FiberNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG (no interaction)

```{r ref.label='AG15NI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Pre-eclampsia (within T1D women)

```{r ref.label='Preeclampsia', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

##Differential abundance analysis Phylum taxonomic level

**Controlling for multiple measurements, sequencing run, conception age, BMI, parity and HLA type**

```{r ref.label='de1', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='design', echo=FALSE}
```

```{r ref.label='de2', fig.height=7, fig.width=9, echo=FALSE}
```

```{r ref.label='de3', echo=FALSE}
```

###Across all trimesters (T1D vs non-T1D)

```{r ref.label='nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='Counts', echo=FALSE}
```

### Only Trimester 1 (nonT1D vs T1D)

```{r ref.label='T1_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T1_nonT1D_vs_T1D_Subset', echo=FALSE}
```

```{r ref.label='CountsT1', echo=FALSE}
```

### Only Trimester 2 (nonT1D vs T1D)

```{r ref.label='T2_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T2_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CPlotT2', echo=FALSE}
```

```{r ref.label='CountsT2', echo=FALSE}
```

After filtering low prevalent taxa, no species was significantly different between T1D and non-T1D women in this comparison. 

### Only Trimester 3 (nonT1D vs T1D)

```{r ref.label='T3_nonT1D_vs_T1D', echo=FALSE}
```

```{r ref.label='T3_nonT1D_vs_T1D_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsT3', echo=FALSE}
```

### T1 vs T2 (T1D and non-T1D data)

```{r ref.label='T1_vs_T2', echo=FALSE}
```

```{r ref.label='T1_vs_T2_Subset', echo=FALSE}
```

```{r ref.label='CountsTriCorr12', echo=FALSE}
```

### T2 vs T3 (T1D and non-T1D data)

```{r ref.label='T2_vs_T3', echo=FALSE}
```

```{r ref.label='T2_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr23', echo=FALSE}
```

Nothing left after filtering by prevalence and LogFC

### T1 vs T3 (T1D and non-T1D data)

```{r ref.label='T1_vs_T3', echo=FALSE}
```

```{r ref.label='T1_vs_T3_Subset', fig.height=9, fig.width=9, echo=FALSE, echo=FALSE}
```

```{r ref.label='CountsTriCorr13', echo=FALSE}
```

### T1 vs T2 (in women with T1D)

```{r ref.label='T1DT1vsT2', echo=FALSE}
```

```{r ref.label='T1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT12', echo=FALSE}
```

### T2 vs T3 (in women with T1D)

```{r ref.label='T1DT2vsT3', echo=FALSE}
```

```{r ref.label='T1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT23', echo=FALSE}
```

### T1 vs T3 (in women with T1D)

```{r ref.label='T1DT1vsT3', echo=FALSE}
```

```{r ref.label='T1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsT1DT13', echo=FALSE}
```

### T1 vs T2 (in women without T1D)

```{r ref.label='noT1DT1vsT2', echo=FALSE}
```

```{r ref.label='noT1DT1vsT2_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT12', echo=FALSE}
```

### T2 vs T3 (in women without T1D)

```{r ref.label='noT1DT2vsT3', echo=FALSE}
```

```{r ref.label='noT1DT2vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT23', echo=FALSE}
```

### T1 vs T3 (in women without T1D)

```{r ref.label='noT1DT1vsT3', echo=FALSE}
```

```{r ref.label='noT1DT1vsT3_PH1', echo=FALSE}
```

```{r ref.label='CountsnonT1DT13', echo=FALSE}
```

