---
title: "Analysis of the gut microbiome during pregnancy in T1D (Alpha and beta diversity analyses)"
author: "Alexandra Roth Schulze"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    number_sections: yes
    toc_float: yes
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
    code_folding: "hide"
urlcolor: blue
editor_options: 
  chunk_output_type: console
---


```{r D8, echo=FALSE}
options(width = 90)
```

```{r D2, include=FALSE}
library(knitr)
opts_chunk$set(comment = NA, warning=FALSE, message=FALSE, echo=TRUE, tidy=TRUE, tidy.opts=list(width.cutoff=60))
library("ggplot2")
theme_set(theme_bw())
pal = "Set1"
scale_colour_discrete <- function(palname = pal, ...) {
    scale_colour_brewer(palette = palname, ...)
}
scale_fill_discrete <- function(palname = pal, ...) {
    scale_fill_brewer(palette = palname, ...)
}
```


```{r PackageLoad, tidy=TRUE, message=FALSE, warning=FALSE, echo=FALSE}
library("data.table")
library("dplyr")
library("ggplot2")
library("ade4")
library("plotly")
library("RColorBrewer")
library("matrixStats")
library("knitr")
library("RColorBrewer")
library("limma")
library("edgeR")
library("phyloseq")
library("vegan")
library("reshape2")
library("scales")
library("lme4")
library("lmerTest")
library("geepack")
library("Hmisc")
library("emmeans")
library("doBy")
library("microbiomeSeq")
library("devtools")
library("adespatial")
library("genefilter")
library("metagMisc")
library("easyGgplot2")
#library("plyr")
```

```{r RMAPfunction, echo=FALSE}
## Function to perform PERMANOVA analysis with repeat measure-aware permutations 
PERMANOVA_repeat_measures <- function(
  D, permute_within, blocks = NULL, block_data, permutations=999,
  metadata_order = c(names(permute_within), names(block_data)),
  na.rm=F) {
  
  # Make sure D is a dist object
  if (class(D) != "dist") {
    stop("D must be a dist object")
  }
  
  # Default to free permutations if blocks is not given
  if (!missing(block_data) && is.null(blocks)) {
    stop("blocks must be given if block_data is present")
  } else if (is.null(blocks)) {
    blocks <- rep(1, nrow(permute_within))
    block_data <- as.data.frame(matrix(0, nrow=1, ncol=0))
  } else if (length(unique(blocks)) == 1) {
    warning("blocks only contains one unique value")
  }
  
  # Ensure no metadata overlap between permute_within and block_data
  if (length(intersect(names(permute_within), names(block_data))) > 0) {
    stop("metadata is repeated across permute_within and block_data")
  }
  
  # Ensure that metadata_order only contains stuff in permute_within and block_data
  if(length(setdiff(metadata_order, union(names(permute_within), names(block_data)))) > 0) {
    stop("metadata_order contains metadata not in permute_within and block_data")
  }
  
  # Ensure that the data in permute_within matches that in dist
  ord <- rownames(as.matrix(D))
  if (length(ord) != nrow(permute_within) || length(blocks) != length(ord)) {
    stop("blocks, permute_within, and D are not the same size")
  }
  if (is.null(rownames(permute_within))) {
    warning("permute_within has no rownames - can't verify sample orders")
  } else if (!all(ord == rownames(permute_within))) {
    stop("rownames do not match between permute_within and D")
  }
  
  # Ensure matching between blocks and block_data
  if (any(is.na(blocks))) {
    stop("NAs are not allowed in blocks")
  }
  if (is.factor(blocks)) {
    if (length(levels(blocks)) != nrow(block_data)) {
      stop("block_data does not have as many rows as blocks has levels")
    }
    if (!is.null(rownames(block_data)) && any(rownames(block_data) != levels(blocks))) {
      stop("block_data rownames does not match the levels of blocks")
    }
    # Discard level information
    blocks <- as.numeric(blocks)
  } else if (is.numeric(blocks)) {
    if (blocks < 1 || max(blocks) > nrow(block_data)) {
      stop("Numeric blocks has indices out of range")
    }
  } else if (is.character(blocks)) {
    if (is.null(rownames(block_data)) || !all(blocks %in% rownames(block_data))) {
      stop("blocks does not match the rownames of block_data")
    }
    # Transform to numeric
    blocks <- match(blocks, rownames(block_data))
  } else {
    stop("blocks must be a numeric, factor, or character vector")
  }
  
  # Error out on NA metadata rather than allowing adonis to error out with
  # a totally nonsensical error message
  if (any(is.na(permute_within)) || any(is.na(block_data))) {
    if (na.rm) {
      n_prerm <- length(blocks)
      
      # Remove NAs in block_data
      hasna <- (rowSums(is.na(block_data)) > 0) | (sapply(split(rowSums(is.na(permute_within)) > 0, blocks), mean) == 1)
      block_data <- block_data[!hasna,, drop=F]
      keep <- !hasna[blocks]
      blocks <- cumsum(!hasna)[blocks]
      
      blocks <- blocks[keep]
      permute_within <- permute_within[keep,, drop=F]
      D <- as.matrix(D)[keep, keep]
      # block_data is not subset, as the rows with NAs are no longer referenced in blocks
      
      # Remove NAs in permute_within
      keep <- rowSums(is.na(permute_within)) == 0
      blocks <- blocks[keep]
      permute_within <- permute_within[keep,, drop=F]
      D <- as.dist(D[keep, keep])
      
      if (length(blocks) < ncol(permute_within) + ncol(block_data)) {
        stop(sprintf("After omitting samples with NAs, the number of samples (%d) is less than the number of metadata (%d)",
                     length(blocks), ncol(permute_within) + ncol(block_data)))
      } else if (length(blocks) < n_prerm * 0.5) {
        warning(sprintf("Removed %d samples with NA metadata", n_prerm - length(blocks)))
      }
    } else {
      stop("Some metadata is NA! adonis does not support any NA in the metadata")
    }
  }
  
  # Warn on some suspicious input
  persample <- apply(permute_within, 1, function(x)is.factor(x) && !any(duplicated(x)))
  if (any(persample)) {
    warning(sprintf("%s in permute_within has one DOF per sample.", colnames(permute_within)[which(persample)[1]]))
  }
  if (length(unique(blocks)) < nrow(block_data)) {
    warning("Not all blocks have a sample associated with them. Block permutations will still be performed over the full set of blocks - if this is not desired, subset block_data to only the blocks which appear in the data.")
  }
  if (!any(duplicated(blocks))) {
    warning("blocks contains no duplicated elements")
  }
  
  library(vegan)
  library(permute)
  
  # Test statistic from non-permuted data
  mtdat <- cbind(permute_within, block_data[blocks,,drop=F])
  ad <- adonis(D ~ ., permutations=0, data=mtdat[, metadata_order, drop=F])
  R2 <- ad$aov.tab$R2
  names(R2) <- rownames(ad$aov.tab)
  
  # Permutations
  nullsamples <- matrix(NA, nrow=length(R2), ncol=permutations)
  for (i in seq_len(permutations)) {
    within.i <- shuffle(nrow(permute_within), control=how(blocks=blocks))
    block.i <- sample(seq_len(nrow(block_data)))
    mtdat <- cbind(
      permute_within[within.i,,drop=F],
      block_data[block.i,,drop=F][blocks,,drop=F])
    perm.ad <- adonis(D ~ ., permutations=0, data=mtdat[, metadata_order, drop=F])
    
    nullsamples[,i] <- perm.ad$aov.tab$R2
  }
  
  # For residuals, test the other direction (i.e. p-value of all covariates)
  n <- length(R2)
  R2[n-1] <- 1 - R2[n-1]
  nullsamples[n-1,] <- 1 - nullsamples[n-1,]
  
  # P value calculation similar to adonis's
  exceedances <- rowSums(nullsamples > R2)
  P <- (exceedances + 1) / (permutations + 1)
  
  P[n] <- NA    # No p-values for "Total"
  ad$aov.tab$`Pr(>F)` <- P
  
  return (ad)
}

##Filter abundance
# function to perform pre-filtering
low.count.removal = function(data, percent=0.01)
{
  keep.otu = which(colSums(data)*100/(sum(colSums(data))) > percent)
  data.filter = data[,keep.otu]
  return(list(data.filter = data.filter, keep.otu = keep.otu))
}
```


#Background
This report shows the analysis performed for metagenomics sequencing from the gut microbiome of 66 pregnant women. For each pregnant woman, samples were taken at 1 to 3 different time points during the pregnancy  (i.e. corresponding to trimesters, up until day ~280 when the child is born) to give a total of 134 samples. From the 66 pregnant women, XXX are diabetic (T1D). For some analyses, time points were categorised into trimesters, correspondence being:

Trimester 1 = 0-99 days
Trimester 2 = 100-196 days
Trimester 3 = 196-280 days


#Background

This report shows the analysis performed for metagenomics sequencing from the gut microbiome of 70 pregnancies (36 in women with T1D) from a total of 66 different women. For each pregnant woman, samples were taken at 1 to 3 different time points during the pregnancy (i.e. corresponding to trimesters, up until day ~280 when the child is born) to give a total of 134 samples. From the 32 pregnant women, 16 are diabetic (T1D). For some analyses, time points were categorised into trimesters, correspondence being:

Trimester 1 = 0-99 days Trimester 2 = 100-196 days Trimester 3 = 196-280 days

#Data generation

Samples were shotgun sequenced using the NovaSeq 6000 illumina sequencing machine at the Ramaciotti Centre for Genomics located at the University of New South Wales (UNSW) in Sydney, Australia.

##Processing the sequences: from sequence data to feature table

The data was processed through pipelines developed by the group of Curtis Huttenhower at Harvard Univerisity and Broad Institute. These pipelines belong to the BioBakery.

The raw data was uploaded to the short read archive (SRA) database (https://www.ncbi.nlm.nih.gov/sra/) with accession number PRJNA604850.

KneadData (https://bitbucket.org/biobakery/biobakery/wiki/kneaddata) was used to filter human (“contaminant”) sequences as well as quality controlling the sequences This has to be done for each sample. Example:

    kneaddata –input 4_S1_R1_001.fastq.gz –input 4_S1_R2_001.fastq.gz -db /home/users/allstaff/schulze.a/bin/Kneaddata/Homo_sapiens –trimmomatic /usr/local/bioinfsoftware/trimmomatic/trimmomatic-0.36 –output ./kneaddata_output -t 12

HUMAnN2 (https://bitbucket.org/biobakery/humann2/wiki/Home) was employed to quantify gene families and pathways; and within HUMANn2, Metaphlan was also used to obtain taxonomic profiles. This process is also performed per sample and read pair1 and read pair2 need to be concatenated and saved in the same file prior to running HUMAn2 (i.e. end-pairing relationships are currently not taken into account during HUMAnN2’s alignment steps. This is due to the fact that HUMAnN2 strictly aligns reads to isolated coding sequences: either at the nucleotide level or through translated search. As such, it will frequently be the case that one read (READ1) will map inside a given coding sequence while its mate-pair (READ2) will not [information obtained from the manual]). Example of running HUMAnN2 with a sample:

    humann2 -i NGU5366A1.fastq -o ./Output_HUMAnN2 –threads 28

Running this outputs several files, among those is “NGU5366A1_metaphlan_bugs_list.tsv”, which is a list of the bacterial taxa found in that sample at all taxonomic levels (i.e. stratified data) and their relative normalizaed abundances. The script merge_metaphlan_tables.py was used to merge the MetaPhlAn output from several samples into one table taxa (rows) vs samples (columns) with the table enlisting the relative normalized abundances per sample per taxa. Example:

    merge_metaphlan_tables.py metaphlan_output1.txt metaphlan_output2.txt metaphlan_output3.txt > ./merged_abundance_table_Rel_Abundance.txt

As mentioned above, the abundance table (profile) contains the stratified data. In order to obtain a table that contains the taxa at a specific taxonomic level (e.g. Species), grep was used. Example for species taxonomic level:

    grep ’s__’ merged_abundance_table_Rel_Abundance.txt | grep -v ’t’ > merged_abundance_table_Rel_Abundance_Sp.tsv # -v (exclude) ’t’ will prevent from selecting those lines that have species and strain (’t__’) classification. We just need species in this case.

Example for genus taxonomic level:

    grep ’g__’ merged_abundance_table_Rel_Abundance.txt | grep -v ’t’ | grep -v ’s’ > merged_abundance_table_Rel_Abundance_Genus.tsv

Note: For other taxonomic levels, this process was repeated for each taxonomic level.

This table contains the whole taxonomy for each taxa, from “kindom” to “Species” along with per sample abundances (i.e. counts per million [CPM]). From this table the taxonomy was obtained. The taxa table is then transformed into a biom file in JSON format and the taxonomic and metadata infomation were added with the software biom. Example:

    biom convert -i merged_abundance_table_Rel_Abundance_Sp.tsv -o merged_abundance_table_Rel_Abundance_Sp.biom –table-type=“OTU table” –to-json
    biom add-metadata –sc-separated taxonomy –observation-header OTUID,taxonomy –observation-metadata-fp Species_taxonomy.txt -i merged_abundance_table_Rel_Abundance_Sp.biom -o Pregnancy_NovaSeq_taxonomic_profile_Rel_Species_NV_wTax.biom
    biom add-metadata -i Pregnancy_NovaSeq_taxonomic_profile_Rel_Species_NV_wTax.biom -o Pregnancy_NovaSeq_taxonomic_profile_Rel_Species_NV_wTaxMet.biom –sample-metadata-fp ./metadata.csv

This process was repeated at for each taxonomic level.

The resulting taxa tables (one for each taxonomic level) with taxonomies and metadata were imported into R to process with the Phyloseq (https://github.com/joey711/phyloseq) and other packages. The data were first de-identified (i.e. data relating to the identification of the donor was removed) and R objects were saved (i.e. A total of 6 taxonomic and 3 functional files were generated). Those R objects were used as inputs in the analysis performed in this document. These objects and other files used in this markdown can be found at https://github.com/PapenfussLab/RothSchulze_pregnancy-gut-microbiome-T1D and all the paths needs to be changed in this document in order to reproduce the analysis.

```{r Taxonomy, echo=FALSE}
Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Species_level_phyloseq_object.RData")
SpeciesObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Strain_level_phyloseq_object.RData")
StrainObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Genus_level_phyloseq_object.RData")
GenusObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Family_level_phyloseq_object.RData")
FamilyObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Order_level_phyloseq_object.RData")
OrderObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Phylum_level_phyloseq_object.RData")
PhylumObj <- Mothers_OTU_AFilt1_Preg

#Mothers_OTU_AFilt1_Preg
TLevel<-c("Species", "Strain", "Genus", "Family", "Order", "Phylum")

Vec <- list("Species" = SpeciesObj, "Strain" =StrainObj, "Genus" = GenusObj, "Family"= FamilyObj, "Order"= OrderObj, "Phylum"= PhylumObj)
Vec2 <- list("Species" = SpeciesObj, "Strain" =StrainObj, "Genus" = GenusObj, "Family"= FamilyObj, "Order"= OrderObj, "Phylum"= PhylumObj)


for (i in 1:6) {
Mothers_OTU_AFilt1_Preg <- Vec[[i]]
T1Dcol = c("blue", "red")

#Making sure each factor has the correct class
#This factor corresponds to extraction batch
sample_data(Mothers_OTU_AFilt1_Preg)$SeqRun <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$SeqRun)
#This factor corresponds to gestational age (day into pregnancy in which the sample was taken)
sample_data(Mothers_OTU_AFilt1_Preg)$Days <- as.integer(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$Days))
#This factor corresponds to conception age
sample_data(Mothers_OTU_AFilt1_Preg)$Age_LMP <- as.numeric(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$Age_LMP))
#This factor corresponds to Body Mass Index at conception time
sample_data(Mothers_OTU_AFilt1_Preg)$BMI_conception <- as.numeric(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$BMI_conception))
#This factor corresponds to Human leukocyte antigen (HLA) type of the women
sample_data(Mothers_OTU_AFilt1_Preg)$HLA.6DRML <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$HLA.6DRML)
#This factor corresponds to parity (which refers to if the women already had children before the current pregnancy)
sample_data(Mothers_OTU_AFilt1_Preg)$Nulliparous <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$Nulliparous)
#This factor corresponds to a unique identifier per person (i.e. if the woman contributed samples from different pregnancies, those samples belong to the same motherid)
sample_data(Mothers_OTU_AFilt1_Preg)$motherid <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$motherid)
#This factor corresponds to T1D status
sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus)
#This factor corresponds to trimester
sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)

#This factor corresponds to a composed factor of trimester and T1D status, which is used in the differential abundance analysis.
sample_data(Mothers_OTU_AFilt1_Preg)$TriwT1D <- factor(paste(sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr,sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus,sep="."))

Vec2[[i]] <- Mothers_OTU_AFilt1_Preg
}
```

```{r Format, echo=FALSE}
Mothers_OTU_AFilt1_Preg <- Vec2[[i]]
#Original table saved in another variable to reset other steps
Original_OTU <- Mothers_OTU_AFilt1_Preg

# Subset to have only mothers trimester 1
Mothers_OTU_AFilt1_Preg_T1 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T1"))
Mothers_OTU_AFilt1_Preg_T1 <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_Preg_T1)>0, Mothers_OTU_AFilt1_Preg_T1)
Original_OTU_T1 <- Mothers_OTU_AFilt1_Preg_T1

# Subset to have only mothers trimester 2
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_Preg_T2)>0, Mothers_OTU_AFilt1_Preg_T2)
Original_OTU_T2 <- Mothers_OTU_AFilt1_Preg_T2

# Subset to have only mothers trimester 3 
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_Preg_T3)>0, Mothers_OTU_AFilt1_Preg_T3)
Original_OTU_T3 <- Mothers_OTU_AFilt1_Preg_T3

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("yes"))
Mothers_OTU_AFilt1_PregwT1D <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_PregwT1D)>0, Mothers_OTU_AFilt1_PregwT1D)
Original_OTU_T1D <- Mothers_OTU_AFilt1_PregwT1D

#Subset to have only mothers with T1D
Mothers_OTU_AFilt1_PregwNT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%c("no"))
Mothers_OTU_AFilt1_PregwNT1D <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_PregwNT1D)>0, Mothers_OTU_AFilt1_PregwNT1D)
Original_OTU_NT1D <- Mothers_OTU_AFilt1_PregwNT1D
```

#Bar plots taxonomy

```{r BarPlot, echo=FALSE, fig.height=7, fig.width=14}
Personal_colors <- c("#FF7200", "#FF0000", "#FFAA00", "#FFDD00", "#72d813", "#154f0d", "#06993E", "#06D8C3", "#06B2D8", "#004ECC", "#0300cc", "#6200CC", "#8E00CC", "#C500CC", "#CC0073", "#CC002C", "#BA8857", "#A04620", "#F47A00", "#771155","#AA4488", "#CC99BB", "#114477", "#4477AA", "#77AADD", "#381C00", "#781156","#A51876","#D21E96","#E43FAD", "#117845","#18A55E","#1ED278","#3FE491","#6CEAAB")
Mothers_OTU_AFilt1_Preg <- Vec2[[1]]
#Duplicate phyloseq object to modify names and use for plot
Mothers_OTU_AFilt1_Preg_TaxPlot <- Mothers_OTU_AFilt1_Preg
#Changing sample names for plotting

sample_names(Mothers_OTU_AFilt1_Preg_TaxPlot) <- sample_data(Mothers_OTU_AFilt1_Preg_TaxPlot)$sample_id

p <- plot_taxa(Mothers_OTU_AFilt1_Preg_TaxPlot, grouping_column = "T1Dstatus", method = "hellinger", number.taxa = 25, filename = NULL)
p + scale_fill_manual(values=Personal_colors) + scale_color_manual(values=Personal_colors) + theme(legend.title = element_text(size=15, face="bold"), legend.text = element_text(size=15, face="italic"), axis.text.y =element_text(size=15))  
```

#Alpha diversity

##Alpha diversity (T1D vs. non-T1D continuous time [Days])

```{r DivCalc, echo=FALSE}
#for(i in 1:6){ #Checked at all taxonomic levels and there was no difference between T1D and non-T1D.
Mothers_OTU_AFilt1_Preg <- Vec2[[1]]
##  Calculate observed richness  ###
#Obtain presence/absence counts to calculate number of observed OTUs (this is done because the original OTU table is normalized at relative abundance, decimal numbers)
Mothers_OTU_AFilt1_Preg_PA <- phyloseq_standardize_otu_abundance(Mothers_OTU_AFilt1_Preg, method = "pa")
DivCal_R <- estimate_richness(Mothers_OTU_AFilt1_Preg_PA, measures=c("Observed")) #Checked hist(DivCal_R) and distributes normaly, hence it is ok to use GEE.
# make a data-frame with sample_data 
DivCal_R_df <- data.frame(DivCal_R, sample_data(Mothers_OTU_AFilt1_Preg_PA))
DivCal_Sort <- arrange(DivCal_R_df, Days)

# Mean-centre days, age at LMP and BMI (adjustment variables) so that the coefficients from the models are meaningful
meandays <- with(DivCal_R_df, mean(Days))
DivCal_R_df$days_c <- with(DivCal_R_df, Days - meandays)

meanage <- with(DivCal_R_df, mean(Age_LMP))
DivCal_R_df$age_c <- with(DivCal_R_df, Age_LMP - meanage)

meanbmi <- with(DivCal_R_df, mean(BMI_conception))
DivCal_R_df$bmi_c <- with(DivCal_R_df, BMI_conception - meanbmi)

DivCal_R_df$nullip <-(DivCal_R_df$Nulliparous)
DivCal_R_df$nullip <- factor(DivCal_R_df$nullip, levels=c(0,1), labels=c("No", "Yes"))

DivCal_R_df$HLA <-factor(DivCal_R_df$HLA.6DRML)

DivCal_R_df$Tri <-factor(DivCal_R_df$TriCorr)

DivCal_R_df$t1dfactor <- factor(DivCal_R_df$T1Dstatus, levels=c("no","yes"), labels=c("nonT1D", "T1D"))

ob_gee2 <- geeglm(Observed ~ t1dfactor*days_c + age_c + nullip + bmi_c + HLA, data=DivCal_R_df, id=motherid, corstr="exchangeable")
ResObs2 <- summary(ob_gee2)
print(ResObs2)

#Here I just first checked that there was no interaction of T1Dstatus and trimester rather than days, and there wasn't and secondly I did the test without the interaction and at no taxonomic level was there a difference between T1D and non-T1D women
#ob_gee2 <- geeglm(Observed ~ age_c + nullip + bmi_c + HLA + t1dfactor, data=DivCal_R_df, id=motherid, corstr="exchangeable")
#ResObs2 <- summary(ob_gee2)
#print(ResObs2)

#}
```


##Alpha diversity (time continuous [see T1D vs. non-T1D])

```{r AlphaBoxplotsF, fig.height=4, fig.width=5, echo=FALSE}
ob_gee3 <- geeglm(Observed ~ t1dfactor*Tri + age_c + nullip + bmi_c + HLA, data=DivCal_R_df, id=motherid, corstr="exchangeable")
ResObs3 <- summary(ob_gee3)
ResObs3

sample_data(Mothers_OTU_AFilt1_Preg_PA)$T1Dstatus <- relevel(sample_data(Mothers_OTU_AFilt1_Preg_PA)$T1Dstatus, ref="yes")
#T1Dcol <- c("blue", "red") #blue: non-T1D, red: T1D
T1Dcol <- c("red", "blue") #blue: non-T1D, red: T1D

Rich_R_OTU <- plot_richness(Mothers_OTU_AFilt1_Preg_PA, x="T1Dstatus", scales = "fixed", measures=c("Observed"))
Rich_R_OTU_P <-  Rich_R_OTU + geom_boxplot(data=Rich_R_OTU$data, aes(color=T1Dstatus), alpha=0.1, size=1)  +  theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + labs(color="T1D status")  + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol) +  theme(legend.position='none')
Rich_R_OTU_P

Rich_R_OTU <- plot_richness(Mothers_OTU_AFilt1_Preg_PA, x="TriCorr", scales = "fixed", measures=c("Observed"))
Rich_R_OTU_P <-  Rich_R_OTU + geom_boxplot(data=Rich_R_OTU$data, aes(color=TriCorr), alpha=0.1, size=1)  +  theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + labs(color="Trimester")   +  theme(legend.position='none')
Rich_R_OTU_P

```

#**Species taxonomic level**

##Microbial community composition analysis at **Species** level (Beta diversity analysi)

**Hypothesis**

Hypothesis: the gut microbiome taxonomic composition (i.e. Beta diversity) during pregnancy differs between women with and without T1D. In order to test if this hypothesis holds true, a measurement of the distance between each pair of samples is calculated (i.e. Bray-Curtis) and a repeated measure aware permutational analysis of variance (i.e. RMA-PERMANOVA) test is applied. A P-value <0.05 is considered to be significant, meaning that our hypothesis cannot be rejected. Borderline P-values are also considered positive.

**Problems with the available R function for Beta diversity analysis**

The current available function for performing hypothesis testing of differences in the microbiome composition between groups of samples is called Adonis (i.e. which perform a PERMANOVA test) and is part of the Rpackage **vegan** (Oksanen, J. et al). The main problem with this function is that if the metadata of interest does not vary with time (e.g. disease status, sex, etc.), adonis does not calculate the corresponding P value correctly, as it permutes levels within a subject. This does not make sense for something like disease status or sex as permuting within-subject will produce the exact same distribution each time you permute. This a known drawback of the adonis function for repeated measures.  


In order to get the correct P-value for a time-invariant metadata, the permutation procedure has to be altered such that it permutes the subjects rather than levels within subjects, which apply to our current pregnancy dataset. 


Here I'm running a script written by **Jason Lloyd-Price** from Curtis Huttenhower lab in Harvard. This script is being used instead of the regular Adonis (PERMANOVA test) from the R package Vegan because as stated above, our metadata of interest (i.e. T1D status) does not vary with time, we  have a mixture of data which changes within and between an individual and we also have unequal group sizes (i.e. we do not always have exactly 3 samples per woman). For those three reasons, it was recommended to use his script to perform the RMA-PERMANOVA analysis.


**PARAMETERS:**


**permute_within:** This data frame has samples on rows and metadata on columns. This should only contain metadata that varies within a block (i.e. it's a single-column data frame with only time/Days).*Since, the other metadata within the block should be the same between repeated measurements and this will not hold when we introduce other factor i.e. sequencing run, BMI, parity, gestational age (i.e. Days or trimester) and conception age , those five factors should be placed here*


**blocks**: This should just be a vector giving the group of each sample (i.e. the motherid vector; motherid is strictly accounting for personID but not for pregnancy per se). 


**block_data**: This data frame contains per-block metadata, with one row per block (motherid). It should only contain metadata pertaining to the blocks (i.e. T1Dstatus). *motherid is not numeric, ensure that the row names match the factor names in the motherid vector (blocks)*.

**metadata_order**: This is needed if you want to specify a particular order that the model should be fit in. Metadata earlier in the list will be fit and residualized first, so these should be **features we are NOT interested in and want to control for**. 


**Results from PERMANOVA test with repeat measure-aware permutations - Controlling for Days/Trimester, sequencing run, conception age, BMI, parity and HLA type**

Here the blocking factor is taking into account each mother who might have given samples from different trimesters or two different pregnancies (i.e. motherid is the name of the factor). Therefore, the factor "motherid" takes into account two different pregnancies from the same mother as one SubjectID to adjust for repeated measurements

The interaction between T1D status and time was also included in order to test if the differences between T1D and non-T1D women in the same or changes throguhout pregnancy.

###Interaction between T1D status and Time

```{r Seed1, echo=FALSE}
set.seed(201)
i=1
```

```{r ref.label='Format', echo=FALSE}
```

```{r DistMatMetadata, echo=FALSE}
#Calculating bray-Curtis distace matrix
Mothers_OTU_AFilt1_Preg <- transform_sample_counts(Original_OTU, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg)))
```

```{r RMAP, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Days <- as.numeric(as.character(Meta_df$Days))
PW$Days <- Days
PW$T1D_Time_Interaction <- (Meta_df$T1Dstatus == "yes") * PW$Days ## To test the interaction as a continuous value aka Days!
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r RMAPeval, echo=FALSE}
#Calling RMA-PERMANOVA function
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub)
```

```{r ResultRMAP, echo=FALSE}
#P-value for interaction of T1D and time
RMAPres
RT1DTimeInt <- round(RMAPres$aov.tab$`Pr(>F)`[3], 4)
RT1DTimeInt
```

Due to the interaction between T1D and time having a significant P-value, differences in beta diversity between T1D and non-T1D was assessed by trimester using a normal PERMANOVA with the adonis function.


```{r PlotT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus", shape="TriCorr")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  
p_OTU12pr
```

###T1D status in trimester 1

```{r DistMatxT1, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T1 <- transform_sample_counts(Original_OTU_T1, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T1, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T1)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
Meta_dfTri$HLA.6DRML <- as.factor(Meta_dfTri$HLA.6DRML)
```

```{r RMAPxTri, echo=FALSE}
AdonisR <- adonis(D ~  SeqRun + Nulliparous + Age_LMP + BMI_conception + HLA.6DRML + T1Dstatus, data=Meta_dfTri)
AdonisR
```

```{r ResultRMAPAdonisT1D, echo=FALSE}
#P-value for T1D and time
RT1D <- round(AdonisR$aov.tab$`Pr(>F)`[6], 4)
RT1D
```

**Beta diversity Plot**

```{r PlotT1DxT1,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T1, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T1, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  +  theme(legend.position='none')# + ggtitle( paste( "P-value =", RT1D ) )  
p_OTU12pr
```

###T1D status in trimester 2

```{r DistMatMetadataT2, echo=FALSE}
#Calculating bray-Curtis distace matrix
Mothers_OTU_AFilt1_Preg_T2 <- transform_sample_counts(Original_OTU_T2, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T2, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T2)))
```

```{r RMAPT2, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus

RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub)
RMAPres
```

**Beta diversity Plot**

```{r PlotT1DxT2,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T2, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T2, ordination_OTU, type="samples", axes=c(2, 3),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  +  theme(legend.position='none')# + ggtitle( paste( "P-value =", RT1D ) )
p_OTU12pr
```

###T1D status in trimester 3

```{r DistMatMetadataT3, echo=FALSE}
#Calculating bray-Curtis distace matrix
Mothers_OTU_AFilt1_Preg_T3 <- transform_sample_counts(Original_OTU_T3, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T3, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T3)))
```

```{r RMAPT3, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub)
RMAPres
```

**Beta diversity Plot**

```{r PlotT1DxT3,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
T1Dcol =c("blue", "red")
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T3, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T3, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  +  theme(legend.position='none')# + ggtitle( paste( "P-value =", RT1D ) )
p_OTU12pr
```

###Trimester 3 sensitivity analysis

```{r T3Sensitivity,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
TaxLevel <- c("Species", "Strain", "Genus", "Family", "Order", "Phylum")
for(me in 1:6) {
Mothers_OTU_AFilt1_Preg <- Vec2[[me]]

T1 <-  subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr=="T1")
T1N <- sample_data(T1)$motherid
T3_T1subset <- subset_samples(Mothers_OTU_AFilt1_Preg, motherid%in%T1N)
#Remove 5468_1_T3 since 5468_1_T1 is not in the T1 dataset
T3_T1subset <-  subset_samples(T3_T1subset, sample_id!="5468_1_T3")
T3_T1subset <-  subset_samples(T3_T1subset, TriCorr=="T3")
T3_T1subset <- prune_taxa(taxa_sums(T3_T1subset)>0, T3_T1subset)

T3_T1subsetT3 <- transform_sample_counts(T3_T1subset, function(x) log(1 + x) )
D <- distance(T3_T1subsetT3, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(T3_T1subsetT3)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
Meta_dfTri$HLA.6DRML <- as.factor(Meta_dfTri$HLA.6DRML)
AdonisR <- adonis(D ~  SeqRun + Nulliparous + Age_LMP + BMI_conception + HLA.6DRML + T1Dstatus, data=Meta_dfTri)
AdonisR


cat("For",  TaxLevel[me], "P-T1Dstatus=", AdonisR$aov.tab$`Pr(>F)`[6])
cat('\n')
}
```

###Days within T1D

```{r ChwT1D, echo=FALSE}
Mothers_OTU_AFilt1_PregwT1D <- transform_sample_counts(Original_OTU_T1D, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_PregwT1D, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregwT1D))
Meta_df <- as.data.frame(Meta)
```

```{r RMAPwT1D, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Days <- as.numeric(as.character(Meta_df$Days))
PW$Days <- Days

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r RMAPevaDays, echo=FALSE}
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub, 
    metadata_order = c(names(Meta_sub), names(PW)))
```

```{r ResultRMAPwT1D, echo=FALSE}
RMAPres
RDays <- round(RMAPres$aov.tab$`Pr(>F)`[6], 4)
```

**Beta diversity Plot**

```{r BCPCoADayswT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwT1D, method="PCoA", distance="bray")
mid<-mean(sample_data(Mothers_OTU_AFilt1_PregwT1D)$Days)
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="Days")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RDays ) ) + scale_color_gradient2(midpoint=161, low="blue", mid="white", high="red")
p_OTU12pr
```

###Trimester within T1D

```{r RMAPTriwT1D, echo=FALSE}
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ResultRMAP2wT1D, echo=FALSE}
RMAPres
RTri <- round(RMAPres$aov.tab$`Pr(>F)`[6], 4)
```

**Beta diversity Plot**

```{r BCPCoATriwT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwT1D, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwT1D, ordination_OTU, type="samples", axes=c(2, 4),  color="TriCorr")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  #+ ggtitle( paste( "P-value =", RTri ) )
p_OTU12pr
```

###Days within non-T1D

```{r ChDayswNT1D, echo=FALSE}
Mothers_OTU_AFilt1_PregwNT1D <- transform_sample_counts(Original_OTU_NT1D, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_PregwNT1D, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregwNT1D))
Meta_df <- as.data.frame(Meta)
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r BCPCoADayswNT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwNT1D, method="PCoA", distance="bray")
mid<-mean(sample_data(Mothers_OTU_AFilt1_PregwNT1D)$Days)
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwNT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="Days")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RDays ) ) + scale_color_gradient2(midpoint=161, low="blue", mid="white", high="red")
p_OTU12pr
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r BCPCoATriwNT1D,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwNT1D, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwNT1D, ordination_OTU, type="samples", axes=c(2, 4),  color="TriCorr")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RTri ) )
p_OTU12pr
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r Tri1OF, echo=FALSE}
TOF <- "T1" 
```

```{r Age, echo=FALSE}
Mothers_OTU_AFilt1_Preg <- transform_sample_counts(Original_OTU, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Age_Time_Interaction <- (Meta_df$TriCorr == TOF) * PW$Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r RMAPevaOFactors, echo=FALSE}
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub, 
    metadata_order = c(names(Meta_sub), names(PW)))
RMAPres
```

#####Age at conception interaction with time (reference: trimester 2)

```{r Tri2OF, echo=FALSE}
TOF <- "T2" 
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r Tri3OF, echo=FALSE}
TOF <- "T3" 
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r AgeNI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction: trimester 1)

```{r DistMatxT1Age, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T1 <- transform_sample_counts(Original_OTU_T1, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T1, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T1)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
Meta_dfTri$HLA.6DRML <- as.factor(Meta_dfTri$HLA.6DRML)
```

```{r RMAPxAge, echo=FALSE}
AdonisR <- adonis(D ~  SeqRun + Nulliparous + BMI_conception + HLA.6DRML + T1Dstatus + Age_LMP, data=Meta_dfTri)
AdonisR
```

#####Age at conception (no interaction: trimester 2)

```{r AgeT2, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T2 <- transform_sample_counts(Original_OTU_T2, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T2, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T2))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Age at conception (no interaction: trimester 3)

```{r AgeT3, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T3 <- transform_sample_counts(Original_OTU_T3, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T3, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T3))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r BMI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$BMI_Time_Interaction <- (Meta_df$TriCorr == TOF) * PW$BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r BMINI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r HLArefDR34, echo=FALSE}
HLAr <- "DR34"
```

```{r HLA, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$HLA_Time_Interaction <- (Meta_df$HLA.6DRML == HLAr) * PW$Days #FOR HLA DR34 DRXX Group3o4

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r HLArefDRXX, echo=FALSE}
HLAr <- "DRXX"
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r HLArefDR3o4, echo=FALSE}
HLAr <- "Group3o4"
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r HLANI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
T1Dstatus <- Meta_subj$T1Dstatus
Meta_sub <- as.data.frame(T1Dstatus)
rownames(Meta_sub) <- Meta_subj$motherid
HLA <- as.factor(Meta_subj$HLA.6DRML)
Meta_sub$HLA <- HLA
```

```{r RMAPevaOFactorsNI, echo=FALSE}
RMAPres <- PERMANOVA_repeat_measures(
    D = D, permutations=999,
    permute_within=PW,
    blocks=BS, block_data=Meta_sub)
RMAPres
```

#####HLA (no interaction: T1)

```{r DistMatxT1HLA, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T1 <- transform_sample_counts(Original_OTU_T1, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T1, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T1)))
Meta_df$Age_LMP <- as.numeric(as.character(Meta_df$Age_LMP))
Meta_df$BMI_conception <- as.numeric(as.character(Meta_df$BMI_conception))
Meta_df$HLA.6DRML <- as.factor(Meta_df$HLA.6DRML)
```

```{r RMAPxHLA, echo=FALSE}
AdonisR <- adonis(D ~  SeqRun + Nulliparous + BMI_conception  + T1Dstatus + Age_LMP + HLA.6DRML, data=Meta_df)
AdonisR
```

#####HLA type (no interaction: T2)

```{r HLANIT2, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T2 <- transform_sample_counts(Original_OTU_T2, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T2, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T2))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
T1Dstatus <- Meta_subj$T1Dstatus
Meta_sub <- as.data.frame(T1Dstatus)
rownames(Meta_sub) <- Meta_subj$motherid
HLA <- as.factor(Meta_subj$HLA.6DRML)
Meta_sub$HLA <- HLA
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####HLA type (no interaction: T3)

```{r HLANIT3, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T3 <- transform_sample_counts(Original_OTU_T3, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T3, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T3))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
T1Dstatus <- Meta_subj$T1Dstatus
Meta_sub <- as.data.frame(T1Dstatus)
rownames(Meta_sub) <- Meta_subj$motherid
HLA <- as.factor(Meta_subj$HLA.6DRML)
Meta_sub$HLA <- HLA
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r Parity, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Parity_Time_Interaction <- (PW$Parity == "1") * PW$Days

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction)

```{r ParityNI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r MOD.CER, echo=FALSE}
MOD <- "CER"
```

```{r MOD, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$MOD_Time_Interaction <- (Meta_df$MOD == MOD) * PW$Days #FOR CER CE and LAB


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
MOD <- as.factor(Meta_subj$MOD)
Meta_sub$MOD <- MOD
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r MOD.CE, echo=FALSE}
MOD <- "CE"
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r MOD.Lab, echo=FALSE}
MOD <- "LAB"
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction)

```{r MODNI, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg, method="bray")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
MOD <- as.factor(Meta_subj$MOD)
Meta_sub$MOD <- MOD
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Mode of delivery (no interaction: T1)

```{r DistMatxT1MOD, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T1 <- transform_sample_counts(Original_OTU_T1, function(x) log(1 + x) )
D <- distance(Mothers_OTU_AFilt1_Preg_T1, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T1)))
Meta_df$Age_LMP <- as.numeric(as.character(Meta_df$Age_LMP))
Meta_df$BMI_conception <- as.numeric(as.character(Meta_df$BMI_conception))
Meta_df$HLA.6DRML <- as.factor(Meta_df$HLA.6DRML)
Meta_df$MOD <- as.factor(Meta_df$MOD)
```

```{r RMAPxMOD, echo=FALSE}
AdonisR <- adonis(D ~  SeqRun + Nulliparous + BMI_conception  + T1Dstatus + Age_LMP + HLA.6DRML + MOD, data=Meta_df)
AdonisR
```

#####Mode of delivery (no interaction: T2)

```{r MOT2, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg_T2, method="bray")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T2))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
MOD <- as.factor(Meta_subj$MOD)
Meta_sub$MOD <- MOD
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Mode of delivery (no interaction: T3)

```{r MOT3, echo=FALSE}
D <- distance(Mothers_OTU_AFilt1_Preg_T3, method="bray")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T3))
Meta_df <- as.data.frame(Meta)
#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
MOD <- as.factor(Meta_subj$MOD)
Meta_sub$MOD <- MOD
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r Carbohydrate, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, carbohydrate_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$carbohydrate_g <- as.numeric(as.character(Meta_df$carbohydrate_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Carb_Time_Interaction <- (Meta_df$TriCorr == TOF) * Meta_df$carbohydrate_g

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Carbs <- as.numeric(as.character(Meta_subj$carbohydrate_g)) 
Meta_sub$Carbs <- Carbs
```

```{r Seed1.1, echo=FALSE}
set.seed(711)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r CarbohydrateNI, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, carbohydrate_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$carbohydrate_g <- as.numeric(as.character(Meta_df$carbohydrate_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Carbs <- as.numeric(as.character(Meta_subj$carbohydrate_g)) 
Meta_sub$Carbs <- Carbs
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r Fibre, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, fibre_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$fibre_g <- as.numeric(as.character(Meta_df$fibre_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
PW$Fiber_Time_Interaction <- (Meta_df$TriCorr == TOF) * Meta_df$fibre_g

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Fiber <- as.numeric(as.character(Meta_subj$fibre_g)) 
Meta_sub$Fiber <- Fiber
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r FibreNI, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregC <- subset_samples(Mothers_OTU_AFilt1_Preg, fibre_g!= "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregC))
D <- distance(Mothers_OTU_AFilt1_PregC, method="bray")
Meta_df <- as.data.frame(Meta)
Meta_df$fibre_g <- as.numeric(as.character(Meta_df$fibre_g ))
#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Tri <- as.factor(Meta_df$TriCorr)
PW$Tri <- Tri
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
Fiber <- as.numeric(as.character(Meta_subj$fibre_g)) 
Meta_sub$Fiber <- Fiber
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r AG15, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregG <- subset_samples(Mothers_OTU_AFilt1_Preg, value_15agMother != "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregG))
Meta_df <- as.data.frame(Meta)
D <- distance(Mothers_OTU_AFilt1_PregG, method="bray")
Meta_df$value_15agMother <- as.numeric(as.character(Meta_df$value_15agMother))

#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
AG15 <- as.numeric(as.character(Meta_df$value_15agMother))
PW$AG15 <- AG15
PW$AG15_T1D_Interaction <- (Meta_df$T1Dstatus == "yes") * PW$AG15


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r T1DG, echo=FALSE}
T1DG <- "yes"
```

```{r AG15T1D, echo=FALSE}
##Filtering out sample with no information about carbohydrate intake
Mothers_OTU_AFilt1_PregGT1D <- subset_samples(Mothers_OTU_AFilt1_Preg, T1Dstatus%in%T1DG)
Mothers_OTU_AFilt1_PregG <- subset_samples(Mothers_OTU_AFilt1_PregGT1D, value_15agMother != "NA")
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregG))
Meta_df <- as.data.frame(Meta)
D <- distance(Mothers_OTU_AFilt1_PregG, method="bray")
Meta_df$value_15agMother <- as.numeric(as.character(Meta_df$value_15agMother))

#permute_within
Days <- as.numeric(as.character(Meta_df$Days))
PW <- as.data.frame(Days)
rownames(PW) <- rownames(Meta_df)
seqRun <- as.factor(Meta_df$SeqRun)
PW$seqRun <- seqRun
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
AG15 <- as.numeric(as.character(Meta_df$value_15agMother))
PW$AG15 <- AG15


#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r NT1DG, echo=FALSE}
T1DG <- "no"
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed1.2, echo=FALSE}
set.seed(201)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#**Strain taxonomic level**

##Microbial community composition analysis at **strain** level (Beta diversity analysis)

```{r i2, echo=FALSE}
i=2
```

```{r ref.label='Format', echo=FALSE}
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r Seed1.3, echo=FALSE}
set.seed(48)
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction and within trimester 1)

```{r T1NI, echo=FALSE}
TNI <- "T1"
```

```{r AgeNITri, echo=FALSE}
Mothers_OTU_AFilt1_PregT1Age <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%TNI)
D <- distance(Mothers_OTU_AFilt1_PregT1Age, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregT1Age))
Meta_df <- as.data.frame(Meta)

#permute_within
seqRun <- as.factor(Meta_df$SeqRun)
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age

#blocks
BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction and within trimester 2)

```{r T2NI, echo=FALSE}
TNI <- "T2"
```

```{r ref.label='AgeNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction and within trimester 3)

```{r T3NI, echo=FALSE}
TNI <- "T3"
```

```{r ref.label='AgeNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r BMINITri, echo=FALSE}
Mothers_OTU_AFilt1_PregT1Par <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%TNI)
D <- distance(Mothers_OTU_AFilt1_PregT1Par, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregT1Par))
Meta_df <- as.data.frame(Meta)
#permute_within

seqRun <- as.numeric(as.character(Meta_df$SeqRun))
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='BMINITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='BMINITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ParityNITri, echo=FALSE}
Mothers_OTU_AFilt1_PregT1Par <- subset_samples(Mothers_OTU_AFilt1_Preg, TriCorr%in%TNI)
D <- distance(Mothers_OTU_AFilt1_PregT1Par, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Mothers_OTU_AFilt1_PregT1Par))
Meta_df <- as.data.frame(Meta)
#permute_within

seqRun <- as.numeric(as.character(Meta_df$SeqRun))
PW <- as.data.frame(seqRun)
rownames(PW) <- rownames(Meta_df)
Age <- as.numeric(as.character(Meta_df$Age_LMP))
PW$Age <- Age
BMI <- as.numeric(as.character(Meta_df$BMI_conception))
PW$BMI <- BMI
Parity <- Meta_df$Nulliparous
PW$Parity <- Parity

BS <- as.vector(Meta_df$motherid)
#block_data
Meta_subj <- Meta_df[order(Meta_df$motherid),]
Meta_subj <- Meta_subj[!duplicated(Meta_subj$motherid),, drop=F]
HLA <- Meta_subj$HLA.6DRML
Meta_sub <- as.data.frame(HLA)
rownames(Meta_sub) <- Meta_subj$motherid
T1Dstatus <- as.factor(Meta_subj$T1Dstatus)
Meta_sub$T1Dstatus <- T1Dstatus
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction)

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r ref.label='CarbohydrateNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r ref.label='FibreNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed1.4, echo=FALSE}
set.seed(87)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#**Genus taxonomic level**

##Microbial community composition analysis at **Genus** level (Beta diversity analysis)

```{r i3, echo=FALSE}
i=3
```

```{r ref.label='Format', echo=FALSE}
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r Seed2, echo=FALSE}
set.seed(43)
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time  (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction)

```{r ref.label='ParityNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction)

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r ref.label='CarbohydrateNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r ref.label='FibreNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed2.5, echo=FALSE}
set.seed(711)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed2.6, echo=FALSE}
set.seed(888)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#**Family taxonomic level**

##Microbial community composition analysis at **Family** level (Beta diversity analysis)

```{r i4, echo=FALSE}
i=4
```

```{r ref.label='Format', echo=FALSE}
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r Seed3, echo=FALSE}
set.seed(37)
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####HLA (no interaction: T1)

```{r ref.label='DistMatxT1HLA', echo=FALSE}
```

```{r ref.label='RMAPxHLA', echo=FALSE}
```

#####HLA type (no interaction: T2)

```{r ref.label='HLANIT2', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####HLA type (no interaction: T3)

```{r ref.label='HLANIT3', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction)

```{r ref.label='ParityNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction)

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r ref.label='CarbohydrateNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r ref.label='FibreNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed3.4, echo=FALSE}
set.seed(67)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#**Order taxonomic level**

##Microbial community composition analysis at **Order** level (Beta diversity analysis)

```{r i5, echo=FALSE}
i=5
```

```{r ref.label='Format', echo=FALSE}
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r Seed3.5, echo=FALSE}
set.seed(48)
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r Seed3.6, echo=FALSE}
set.seed(711)
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction)

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Mode of delivery (no interaction: T1)

```{r ref.label='DistMatxT1MOD', echo=FALSE}
```

```{r ref.label='RMAPxMOD', echo=FALSE}
```

#####Mode of delivery (no interaction: T2)

```{r ref.label='MOT2', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Mode of delivery (no interaction: T3)

```{r ref.label='MOT3', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r ref.label='CarbohydrateNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r ref.label='FibreNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed3.7, echo=FALSE}
set.seed(567)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#**Phylum taxonomic level**

##Microbial community composition analysis at **Phylum** level (Beta diversity analysis)

```{r i6, echo=FALSE}
i=6
```

```{r ref.label='Format', echo=FALSE}
```

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadata', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r Seed4, echo=FALSE}
set.seed(48)
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1D', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Other factors

####Age at conception

#####Age at conception interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Age', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Age at conception (no interaction)

```{r ref.label='AgeNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Body max index (BMI)

#####BMI interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####BMI interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='BMI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```
 
#####BMI (no interaction)

```{r ref.label='BMINI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####HLA type

#####HLA type interaction with time (reference: DR34 )

```{r ref.label='HLArefDR34', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: DRXX)

```{r ref.label='HLArefDRXX', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type interaction with time (reference: Group3o4)

```{r ref.label='HLArefDR3o4', echo=FALSE}
```

```{r ref.label='HLA', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####HLA type (no interaction)

```{r ref.label='HLANI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Parity

#####Parity interaction with time

```{r ref.label='Parity', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T1)

```{r ref.label='T1NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T2)

```{r ref.label='T2NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Parity (no interaction but within T3)

```{r ref.label='T3NI', echo=FALSE}
```

```{r ref.label='ParityNITri', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

####Mode of delivery

#####Mode of delivery interaction with time (reference: Emergency caesarea)

```{r ref.label='MOD.CER', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Elective caesarea)

```{r ref.label='MOD.CE', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery interaction with time (reference: Labour)

```{r ref.label='MOD.Lab', echo=FALSE}
```

```{r ref.label='MOD', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Mode of delivery (no interaction)

```{r ref.label='MODNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Mode of delivery (no interaction: T1)

```{r ref.label='DistMatxT1MOD', echo=FALSE}
```

```{r ref.label='RMAPxMOD', echo=FALSE}
```

#####Mode of delivery (no interaction: T2)

```{r ref.label='MOT2', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

#####Mode of delivery (no interaction: T3)

```{r ref.label='MOT3', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Carbohydrate intake

#####Carbohydrate interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Carbohydrate', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Carbohydrate (no interaction)

```{r ref.label='CarbohydrateNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####Fibre intake

#####Fibre interaction with time (reference: trimester 1)

```{r ref.label='Tri1OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 2)

```{r ref.label='Tri2OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre interaction with time (reference: trimester 3)

```{r ref.label='Tri3OF', echo=FALSE}
```

```{r ref.label='Fibre', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####Fibre (no interaction)

```{r ref.label='FibreNI', echo=FALSE}
```

```{r ref.label='RMAPevaOFactorsNI', echo=FALSE}
```

####1,5-AG (glucose control measurement)

#####1,5-AG interaction with time

```{r ref.label='AG15', echo=FALSE}
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG within T1D women

```{r ref.label='T1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed4.5, echo=FALSE}
set.seed(99)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#####1,5-AG interaction with time (within non-T1D women)

```{r ref.label='NT1DG', echo=FALSE}
```

```{r ref.label='AG15T1D', echo=FALSE}
```

```{r Seed4.6, echo=FALSE}
set.seed(789)
```

```{r ref.label='RMAPevaOFactors', echo=FALSE}
```

#Functional analysis (Pathways)

In order to functionally annotate the metagenomic data, we used Kraken2 to filter out human genes more stringently before running HUMAnN2 again. For each sample, the HUMAnN2 output contained the following files:

* XX_pathcoverage.tsv: This file details the abundance of each **pathway** in the community as a function of the abundances of the pathway's component reactions, with each reaction's abundance computed as the sum over abundances of genes catalyzing the reaction. Pathways with zero abundance are not included in the file. Pathway abundance is proportional to the number of complete "copies" of the pathway in the community (from the HUMAnN2 manual). 

* XX_genefamilies.tsv: This file details the abundance of each gene family in the community. Gene families are groups of evolutionarily-related protein-coding sequences that often perform similar functions. Gene family abundance at the community level is stratified to show the contributions from known and unknown species. Individual species' abundance contributions sum to the community total abundance (from the HUMAnN2 manual).

1) The genefamilies.csv file can contain a very large number of features depending on the complexity of your underlying sample. In order to simplify the exploration of gene family abundance data, users can regroup gene families into other functional categories (from the HUMAnN2 manual). Therefore, we regrouped our resulting gene families into **MetaCyc** reactions and **Kegg Orthology (KO)** functional categories, which has to be done in a per sample manner. Example:

  * humann2_regroup_table --input NGU5366A1_genefamilies.tsv --groups uniref90_ko --output ./NGU5366A1_genefamilies_Regrouped_KO.tsv

2) HUMAnN2 quantifies genes and pathways in units of RPKs (reads per kilobase). These account for gene length but not sample sequencing depth. Therefore, a user will renormalize their samples prior to downstream analysis to relative abundance or copies per million (CPM) units. Both of these represent "total sum scaling (TSS)"-style normalization: in the former case, each sample is constrained to sum to 1, whereas in the latter case (CPMs) samples are constrained to sum to 1 million. Units out of 1 million are often more convenient for tables with many, many features (such as genefamilies.tsv tables). CPM as used here does not refer to unnormalized COUNTS per million, but rather copies per million. CPMs as used here are a generic analog of the TPM (transcript per million) unit in RNA-seq. You may wish to use the abbreviation CoPM for added clarity. (from the HUMAnN2 manual). We normilized the counts for each sample file for their three functional tables (XX_genefamilies.tsv, XX_genefamilies_Regrouped_KO.tsv and XX_genefamilies_Regrouped_MetaCyc.tsv) to CoPM. Example:

  * humann2_renorm_table --input NGU5366A1_genefamilies_Regrouped_KO.tsv --output NGU5366A1_genefamilies_Regrouped_KO_CoPM.tsv --units cpm
  
3) Each normalized file, contained one sample. Therefore, in order to obtain a table with the functional profiles for all the samples, we need to join them into one. Example: 

  * humann2_join_tables --input KO --output Pregnancy_genefamilies_KO_Rel.tsv --file_name Complete_genefamilies_Regrouped_KO_Rel.tsv
  
4) The functional profiles are stratified, which means that the sum of the species abundance of a given gene gives the abundance of the gene at the community level. The stratification shows the contributions from known and unknown species to a given function. If we are interested in knowing if there is a significant difference in the abundance of a specific function at the community level (i.e. irrespective of which species is contributing to it), the abundances need to be **unstratified** prior to the analysis. This is done as in the following exmaple:

  * humann2_split_stratified_table --input Complete_genefamilies_Regrouped_KO_Rel.tsv --output Split/

The folder split will contain stratified and unstratified abundances.

Note: Scripts used for regrouping, normalization, joinning tables and stratification are part of HUMAnN2.

The resulting files for pathways, metacyc reactions and Kegg orthology (KO) gene categories were imported into R to process with the Phyloseq (https://github.com/joey711/phyloseq) and other packages. The analysis performed in R is described in this document.

```{r Functions, echo=FALSE}
Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/Pathways_phyloseq_object.RData")
PathwaysObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/KEGG_Orthology_phyloseq_object.RData")
KeggObj <- Mothers_OTU_AFilt1_Preg

Obj <- load("~/Documents/ENDIA Project/Data and analyses/ENDIA Projects analysis/ENDIA_Mothers/Manuscript/Manuscript_Pregnancy_Most_Current/Submission_2021/Code and datasets/MetaCyc_reactions_phyloseq_object.RData")
MetaCycObj <- Mothers_OTU_AFilt1_Preg

#Mothers_OTU_AFilt1_Preg
TLevel<-c("Pathways", "Kegg", "MetaCyc")

Vec <- list("Pathways" = PathwaysObj, "Kegg" =KeggObj, "MetaCyc" = MetaCycObj)
Vec2 <- list("Pathways" = PathwaysObj, "Kegg" =KeggObj, "MetaCyc" = MetaCycObj)


for (i in 1:3) {
Mothers_OTU_AFilt1_Preg <- Vec[[i]]

#Making sure each factor has the correct class
#This factor corresponds to extraction batch
sample_data(Mothers_OTU_AFilt1_Preg)$SeqRun <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$SeqRun)
#This factor corresponds to gestational age (day into pregnancy in which the sample was taken)
sample_data(Mothers_OTU_AFilt1_Preg)$Days <- as.integer(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$Days))
#This factor corresponds to conception age
sample_data(Mothers_OTU_AFilt1_Preg)$Age_LMP <- as.numeric(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$Age_LMP))
#This factor corresponds to Body Mass Index at conception time
sample_data(Mothers_OTU_AFilt1_Preg)$BMI_conception <- as.numeric(as.character(sample_data(Mothers_OTU_AFilt1_Preg)$BMI_conception))
#This factor corresponds to Human leukocyte antigen (HLA) type of the women
sample_data(Mothers_OTU_AFilt1_Preg)$HLA.6DRML <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$HLA.6DRML)
#This factor corresponds to parity (which refers to if the women already had children before the current pregnancy)
sample_data(Mothers_OTU_AFilt1_Preg)$Nulliparous <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$Nulliparous)
#This factor corresponds to a unique identifier per person (i.e. if the woman contributed samples from different pregnancies, those samples belong to the same motherid)
sample_data(Mothers_OTU_AFilt1_Preg)$motherid <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$motherid)
#This factor corresponds to T1D status
sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus)
#This factor corresponds to trimester
sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr <- as.factor(sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr)

#This factor corresponds to a composed factor of trimester and T1D status, which is used in the differential abundance analysis.
sample_data(Mothers_OTU_AFilt1_Preg)$TriwT1D <- factor(paste(sample_data(Mothers_OTU_AFilt1_Preg)$TriCorr,sample_data(Mothers_OTU_AFilt1_Preg)$T1Dstatus,sep="."))

Vec2[[i]] <- Mothers_OTU_AFilt1_Preg
}
```

```{r Pi1, echo=FALSE}
i=1
```

```{r ref.label='Format', echo=FALSE}
```

##Bar plots taxonomy

```{r BarPlotFunc, echo=FALSE, fig.height=7, fig.width=14}
Personal_colors <- c("#FF0000",  "#FF7200", "#FFAA00", "#FFDD00", "#72d813", "#154f0d", "#06993E", "#06D8C3", "#06B2D8", "#004ECC", "#0300cc", "#6200CC", "#8E00CC", "#C500CC", "#CC0073", "#CC002C", "#BA8857", "#A04620", "#F47A00", "#771155","#AA4488", "#CC99BB", "#114477", "#4477AA", "#77AADD", "#781156","#A51876","#D21E96","#E43FAD", "#117845","#18A55E","#1ED278","#3FE491","#6CEAAB", "#381C00")
Mothers_OTU_AFilt1_Preg_Plot <- Mothers_OTU_AFilt1_Preg
sample_names(Mothers_OTU_AFilt1_Preg_Plot) <- sample_data(Mothers_OTU_AFilt1_Preg_Plot)$sample_id
Mothers_OTU_AFilt1_Preg_Rel <- transform_sample_counts(Mothers_OTU_AFilt1_Preg_Plot, function(x) 100 * x/sum(x))
p <- plot_taxa(Mothers_OTU_AFilt1_Preg_Rel, grouping_column = "T1Dstatus", method = "hellinger", number.taxa = 34, filename = NULL)
p + scale_fill_manual(values=Personal_colors) + scale_color_manual(values=Personal_colors)
```

##Functional Alpha diversity (Days) 

```{r DivCalcFunc, echo=FALSE}
##  Calculate observed and estimated richness as well as diversity indexes ###
Mothers_OTU_AFilt1_Preg_PA <- phyloseq_standardize_otu_abundance(Mothers_OTU_AFilt1_Preg, method = "pa")
DivCal_R <- estimate_richness(Mothers_OTU_AFilt1_Preg_PA, measures=c("Observed"))
# make a data-frame with the 5 measures of richness/diversity and sample_data 
sample_data(Mothers_OTU_AFilt1_Preg_PA)$Days <- as.integer(sample_data(Mothers_OTU_AFilt1_Preg_PA)$Days)
DivCal_R_df <- data.frame(DivCal_R, sample_data(Mothers_OTU_AFilt1_Preg_PA))
DivCal_Sort <- arrange(DivCal_R_df, Days)
```

```{r AlphaFunc, echo=FALSE}
meandays <- with(DivCal_R_df, mean(Days))
DivCal_R_df$days_c <- with(DivCal_R_df, Days - meandays)

meanage <- with(DivCal_R_df, mean(Age_LMP))
DivCal_R_df$age_c <- with(DivCal_R_df, Age_LMP - meanage)

meanbmi <- with(DivCal_R_df, mean(BMI_conception))
DivCal_R_df$bmi_c <- with(DivCal_R_df, BMI_conception - meanbmi)

DivCal_R_df$nullip <-(DivCal_R_df$Nulliparous)
DivCal_R_df$nullip <- factor(DivCal_R_df$nullip, levels=c(0,1), labels=c("No", "Yes"))

DivCal_R_df$HLA <-factor(DivCal_R_df$HLA.6DRML)

DivCal_R_df$Tri <-factor(DivCal_R_df$TriCorr)

DivCal_R_df$t1dfactor <- factor(DivCal_R_df$T1Dstatus, levels=c("no","yes"), labels=c("nonT1D", "T1D"))

ob_gee2 <- geeglm(Observed ~ t1dfactor*days_c + age_c + nullip + bmi_c + HLA + SeqRun, data=DivCal_R_df, id=motherid, corstr="exchangeable")
ResObs2 <- summary(ob_gee2)
ResObs2

RT1D <- round(ResObs2$coefficients[2,4], 4)
```

```{r AlphaBoxplot, fig.height=4, fig.width=5, echo=FALSE}
## Observed Richness
T1Dcol <- c("red", "blue")
sample_data(Mothers_OTU_AFilt1_Preg_PA)$T1Dstatus <- relevel(sample_data(Mothers_OTU_AFilt1_Preg_PA)$T1Dstatus, ref="yes")
Rich_R_OTU <- plot_richness(Mothers_OTU_AFilt1_Preg_PA, x="T1Dstatus", scales = "fixed", measures=c("Observed"))
Rich_R_OTU_P <-  Rich_R_OTU + geom_boxplot(data=Rich_R_OTU$data, aes(color=T1Dstatus), alpha=0.1, size=1)  +  theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + labs(color="T1D status") + theme(legend.position='none') + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol) + ggtitle( paste( "P-value =", RT1D ) )
Rich_R_OTU_P
```

##Functional Alpha diversity (Trimesters) 

```{r AlphaTri, echo=FALSE}
ob_gee3 <- geeglm(Observed ~ t1dfactor*Tri + age_c + nullip + bmi_c + HLA + SeqRun, data=DivCal_R_df, id=motherid, corstr="exchangeable")
ResObs3 <- summary(ob_gee3)
ResObs3
```

##Alpha diversity (T1D vs. non-T1D categorical time [Trimester])

```{r AlphaBoxplots, fig.height=4, fig.width=5, echo=FALSE}
T1Dcol <- c("blue", "red") #blue: non-T1D, red: T1D
Rich_R_OTU <- plot_richness(Mothers_OTU_AFilt1_Preg_PA, x="T1Dstatus", scales = "fixed", measures=c("Observed"))
Rich_R_OTU_P <-  Rich_R_OTU + geom_boxplot(data=Rich_R_OTU$data, aes(color=T1Dstatus), alpha=0.1, size=1)  +  theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + labs(color="T1D status")  + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol) +  theme(legend.position='none')
Rich_R_OTU_P
```

##Beta diversity analysis

###Interaction between T1D status and Time

```{r DistMatMetadataF, echo=FALSE}
#Calculating bray-Curtis distace matrix
D <- distance(Original_OTU, method="bray")
# Metadata
Meta_df <- as.data.frame(as.matrix(sample_data(Original_OTU)))
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
#P-value for interaction of T1D and time
```

```{r PlotT1DF,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
Mothers_OTU_AFilt1_Preg <- transform_sample_counts(Original_OTU, function(x) log(1 + x) )
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus", shape="TriCorr")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  
p_OTU12pr
```

###T1D status in trimester 1

```{r DistMatxT1F, echo=FALSE}
D <- distance(Original_OTU_T1, method="bray")
# Metadata
Meta_dfTri <- as.data.frame(as.matrix(sample_data(Original_OTU_T1)))
Meta_dfTri$Age_LMP <- as.numeric(as.character(Meta_dfTri$Age_LMP))
Meta_dfTri$BMI_conception <- as.numeric(as.character(Meta_dfTri$BMI_conception))
Meta_dfTri$HLA.6DRML <- as.factor(Meta_dfTri$HLA.6DRML)
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r PlotT1DxT1F,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T1 <- transform_sample_counts(Original_OTU_T1, function(x) log(1 + x) )
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T1, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T1, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  +  theme(legend.position='none')# + ggtitle( paste( "P-value =", RT1D ) )  
p_OTU12pr
```

###T1D status in trimester 2

```{r DistMatMetadataT2F, echo=FALSE}
#Calculating bray-Curtis distace matrix
Mothers_OTU_AFilt1_Preg_T2 <- subset_samples(Original_OTU_T2, TriCorr%in%c("T2"))
Mothers_OTU_AFilt1_Preg_T2 <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_Preg_T2)>0, Mothers_OTU_AFilt1_Preg_T2)
#D <- distance(Original_OTU_T2, method="bray")
D <- distance(Mothers_OTU_AFilt1_Preg_T2, method="bray")
# Metadata
#Meta_df <- as.data.frame(as.matrix(sample_data(Original_OTU_T2)))
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T2)))
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r PlotT1DxT2F,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T2 <- transform_sample_counts(Original_OTU_T2, function(x) log(1 + x) )
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T2, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T2, ordination_OTU, type="samples", axes=c(1, 2),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  +  theme(legend.position='none')# + ggtitle( paste( "P-value =", RT1D ) )
p_OTU12pr
```

###T1D status in trimester 3

```{r DistMatMetadataT3F, echo=FALSE}
#Calculating bray-Curtis distace matrix
Mothers_OTU_AFilt1_Preg_T3 <- subset_samples(Original_OTU_T3, TriCorr%in%c("T3"))
Mothers_OTU_AFilt1_Preg_T3 <- prune_taxa(taxa_sums(Mothers_OTU_AFilt1_Preg_T3)>0, Mothers_OTU_AFilt1_Preg_T3)
#D <- distance(Original_OTU_T3, method="bray")
D <- distance(Mothers_OTU_AFilt1_Preg_T3, method="bray")
# Metadata
#Meta_df <- as.data.frame(as.matrix(sample_data(Original_OTU_T3)))
Meta_df <- as.data.frame(as.matrix(sample_data(Mothers_OTU_AFilt1_Preg_T3)))
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r PlotT1DxT3F,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
Mothers_OTU_AFilt1_Preg_T3 <- transform_sample_counts(Original_OTU_T3, function(x) log(1 + x) )
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_Preg_T3, method="PCoA", distance="bray")
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_Preg_T3, ordination_OTU, type="samples", axes=c(2, 4),  color="T1Dstatus")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15)) + scale_fill_manual(values=T1Dcol) + scale_color_manual(values=T1Dcol)  +  theme(legend.position='none')# + ggtitle( paste( "P-value =", RT1D ) )
p_OTU12pr
```

###Days within T1D

```{r ChwT1DF, echo=FALSE}
Mothers_OTU_AFilt1_PregwT1D <- transform_sample_counts(Original_OTU_T1D, function(x) log(1 + x) )
D <- distance(Original_OTU_T1D, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Original_OTU_T1D))
Meta_df <- as.data.frame(Meta)
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r BCPCoADayswT1DF,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
Mothers_OTU_AFilt1_PregwT1D <- transform_sample_counts(Original_OTU_T1D, function(x) log(1 + x) )
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwT1D, method="PCoA", distance="bray")
mid<-mean(sample_data(Mothers_OTU_AFilt1_PregwT1D)$Days)
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="Days")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RDays ) ) + scale_color_gradient2(midpoint=161, low="blue", mid="white", high="red")
p_OTU12pr
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ChDayswNT1DF, echo=FALSE}
D <- distance(Original_OTU_NT1D, method="bray")
# Metadata
Meta <- as.matrix(sample_data(Original_OTU_NT1D))
Meta_df <- as.data.frame(Meta)
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r BCPCoADayswNT1DF,  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
#Plot ordination
Mothers_OTU_AFilt1_PregwNT1D <- transform_sample_counts(Original_OTU_NT1D, function(x) log(1 + x) )
ordination_OTU<-ordinate(Mothers_OTU_AFilt1_PregwNT1D, method="PCoA", distance="bray")
mid<-mean(sample_data(Mothers_OTU_AFilt1_PregwNT1D)$Days)
p_OTU12<-plot_ordination(Mothers_OTU_AFilt1_PregwNT1D, ordination_OTU, type="samples", axes=c(1, 2),  color="Days")
p_OTU12pr <- p_OTU12 + geom_point(size = 3)  + theme(axis.title.y = element_text(size=14), axis.title.x = element_text(size=14)) + theme(axis.text.y =element_text(size=15), axis.text.x=element_text(size=15), plot.title = element_text(size=15))  + theme(legend.title = element_text(size=15), legend.text = element_text(size=15))  + ggtitle( paste( "P-value =", RDays ) ) + scale_color_gradient2(midpoint=161, low="blue", mid="white", high="red")
p_OTU12pr
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

#Functional analysis (Kegg Orthology)

```{r Pi2, echo=FALSE}
i=2
```

```{r ref.label='Format', echo=FALSE}
```

##Bar plots taxonomy

```{r ref.label='BarPlotFunc', echo=FALSE, fig.height=7, fig.width=14}
```

##Functional Alpha diversity (Days) 

```{r ref.label='DivCalcFunc', echo=FALSE}
```

```{r ref.label='AlphaFunc', echo=FALSE}
```

```{r ref.label='AlphaBoxplot', fig.height=4, fig.width=5, echo=FALSE}
```

##Functional Alpha diversity (Trimesters) 

```{r ref.label='AlphaTri', echo=FALSE}
```

##Beta diversity analysis

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadataF', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
#P-value for interaction of T1D and time
```

```{r ref.label='PlotT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1F', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1F',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2F', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2F',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3F', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3F',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1DF', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1DF', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

#Functional analysis (MetaCyc reactions)

```{r Pi3, echo=FALSE}
i=3
```

```{r ref.label='Format', echo=FALSE}
```

##Bar plots taxonomy

```{r ref.label='BarPlotFunc', echo=FALSE, fig.height=7, fig.width=14}
```

##Functional Alpha diversity (Days) 

```{r ref.label='DivCalcFunc', echo=FALSE}
```

```{r ref.label='AlphaFunc', echo=FALSE}
```

```{r ref.label='AlphaBoxplot', fig.height=4, fig.width=5, echo=FALSE}
```

##Functional Alpha diversity (Trimesters) 

```{r ref.label='AlphaTri', echo=FALSE}
```

##Beta diversity analysis

###Interaction between T1D status and Time

```{r ref.label='DistMatMetadataF', echo=FALSE}
```

```{r ref.label='RMAP', echo=FALSE}
```

```{r ref.label='RMAPeval', echo=FALSE}
```

```{r ref.label='ResultRMAP', echo=FALSE}
#P-value for interaction of T1D and time
```

```{r ref.label='PlotT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 1

```{r ref.label='DistMatxT1F', echo=FALSE}
```

```{r ref.label='RMAPxTri', echo=FALSE}
```

```{r ref.label='ResultRMAPAdonisT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT1F',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 2

```{r ref.label='DistMatMetadataT2F', echo=FALSE}
```

```{r ref.label='RMAPT2', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT2F',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###T1D status in trimester 3

```{r ref.label='DistMatMetadataT3F', echo=FALSE}
```

```{r ref.label='RMAPT3', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='PlotT1DxT3F',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE}
```

###Days within T1D

```{r ref.label='ChwT1DF', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Days within non-T1D

```{r ref.label='ChDayswNT1DF', echo=FALSE}
```

```{r ref.label='RMAPwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAPwT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoADayswNT1DF',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```

###Trimester within non-T1D

```{r ref.label='RMAPTriwT1D', echo=FALSE}
```

```{r ref.label='RMAPevaDays', echo=FALSE}
```

```{r ref.label='ResultRMAP2wT1D', echo=FALSE}
```

**Beta diversity Plot**

```{r ref.label='BCPCoATriwNT1D',  warning=FALSE, fig.height=4, fig.width=5, echo=FALSE, results='hide'}
```
